<!-- cite_key: khatib2024b -->

# Patient-Centric Knowledge Graphs: A Survey of Current Methods, Challenges, and Applications

Hassan S. Al Khatib, Subash Neupane, Harish Kumar Manchukonda, Noorbakhsh Amiri Golilarz,

Sudip Mittal, Amin Amirlatifi, Shahram Rahimi

Dept. of Computer Science & Engineering, Mississippi State University, Starkville, MS Email: {hsa78, sn922}@msstate.edu, {amin}@che.msstate.edu,{amiri,mittal, rahimi}@cse.msstate.edu

*Abstract*—

Patient-Centric Knowledge Graphs (PCKGs) represent an important shift in healthcare that focuses on individualized patient care by mapping the patient's health information in a holistic and multi-dimensional way. PCKGs integrate various types of health data to provide healthcare professionals with a comprehensive understanding of a patient's health, enabling more personalized and effective care. This literature review explores the methodologies, challenges, and opportunities associated with PCKGs, focusing on their role in integrating disparate healthcare data and enhancing patient care through a unified health perspective. In addition, this review also discusses the complexities of PCKG development, including ontology design, data integration techniques, knowledge extraction, and structured representation of knowledge. It highlights advanced techniques such as reasoning, semantic search, and inference mechanisms essential in constructing and evaluating PCKGs for actionable healthcare insights. We further explore the practical applications of PCKGs in personalized medicine, emphasizing their significance in improving disease prediction and formulating effective treatment plans. Overall, this review provides a foundational perspective on the current state-of-the-art and best practices of PCKGs, guiding future research and applications in this dynamic field.

*Index Terms*—Knowledge Graph, Patient-Centric, Personalized Healthcare, Natural Language Processing, Generative AI

## I. INTRODUCTION

The healthcare industry has experienced a significant shift, transitioning from traditional, provider-centric models towards patient-centered care. This shift highlights the critical role of engaging patients as active participants in their healthcare journeys. At the center of this transformation are PCKGs, which significantly advance personalized, data-driven care. PCKGs facilitate the integration of diverse data types, including medical history, genetics, lifestyle choices, and real-time data from health technology devices, fostering a comprehensive view of patient health essential for customizing treatments to individual needs [\[1\]](#page-15-0) [\[2\]](#page-15-1).

The evolution of Knowledge Graphs (KG) in healthcare into sophisticated networks reflects an increasing acknowledgment of the complexity of human health and the insufficiency of siloed data systems in addressing multifaceted health issues. This evolution is propelled by the necessity for a holistic understanding of patient health, enabling personalized care and applying advanced data analytics to enhance healthcare outcomes [\[3\]](#page-15-2) [\[4\]](#page-15-3). PCKGs stand at the forefront of healthcare innovation, signifying a crucial step towards integrated, patientfocused knowledge networks. This shift from fragmented data systems to cohesive KGs enables healthcare providers to employ machine learning and analytical technologies to help improve precision medicine, diagnostic accuracy, and treatment efficacy. Such a transition represents a technological leap that aligns with the broader objectives of healthcare reform aimed at delivering more personalized, efficient, and patient-centered care [\[5\]](#page-15-4) [\[6\]](#page-15-5). However, incorporating KGs into healthcare presents technical, methodological, and ethical challenges, including data interoperability, privacy concerns, and the complexities of modeling diverse health outcomes. These hurdles pose significant barriers to the widespread adoption of PCKGs, yet the potential of these systems to revolutionize healthcare by offering a nuanced and comprehensive understanding of patient health is unquestionable [\[7\]](#page-15-6) [\[8\]](#page-15-7).

PCKGs' primary aim is to enhance the quality of patient care, improve treatment outcomes, and increase the efficiency of healthcare delivery. By integrating disparate data sources and utilizing advanced analytical models, PCKGs promise to deliver personalized, efficient, and effective healthcare services tailored to each patient's unique needs. This goal emphasizes the shift towards a healthcare system that values and prioritizes individual patient experiences and needs, marking the beginning of a new era of patient-centric, data-driven care [\[9\]](#page-15-8) [\[10\]](#page-15-9). Despite the inherent challenges in integrating KGs into healthcare, the critical need for the advanced application of PCKGs to achieve personalized care and enhance healthcare delivery systems is undeniable. As the healthcare landscape continues to evolve, PCKGs exemplify the industry's commitment to leveraging technology to meet patients' complex and varied needs, thus marking a significant milestone in the journey toward more personalized and effective healthcare solutions.

The motivation of this paper stems from the growing need to consolidate disparate healthcare data into a unified, holistic view for improved patient care. The key contributions of this survey paper are:

- A foundational explanation of knowledge graphs, serving as the theoretical basis for the remainder of the paper.
- Presentation of survey findings and introduction of a taxonomy developed for the field of PCKGs.
- An in-depth review of methodologies specifically designed for PCKGs, shedding light on the most effective techniques currently employed.
- Exploration of real-world applications and use-cases that

have successfully implemented PCKG methodologies, providing evidence of their utility.

The rest of the paper is organized as follows: Section [II](#page-1-0) explains the principles of knowledge graphs, which provide a foundation for the discussions that follow. Section [III](#page-3-0) presents the findings from our survey and introduces the taxonomy we have developed. Moving on to Section [IV,](#page-4-0) we review methodologies explicitly designed for PCKGs. Section [V](#page-10-0) explores real-world applications and examples that benefit from these methodologies. Section [VI](#page-13-0) highlights research challenges and provides targeted recommendations for future scholars in this field. Finally, in Section [VII,](#page-15-10) we summarize the key-findings of this paper and outline directions for work in PCKGs.

### II. BACKGROUND

<span id="page-1-0"></span>The development of knowledge representation has a rich history in the realms of logic and AI. The notion of graphical knowledge representation can be traced back to 1956 when Richens [\[11\]](#page-15-11) introduced the concept of semantic nets. Similarly, symbolic logic knowledge finds its roots in the General Problem Solver [\[12\]](#page-15-12) of 1959. Initially, knowledge bases were employed in knowledge-based systems for reasoning and problem-solving. Notably, MYCIN [\[13\]](#page-15-13), an expert system renowned for medical diagnosis, utilized a knowledge base containing approximately 600 rules. However, it was in 2012 that the concept of Knowledge Graph (KG) gained immense popularity, thanks to Google's search engine and its introduction of the Knowledge Vault framework [\[14\]](#page-15-14). This framework aimed to construct large-scale KGs through knowledge fusion. There is currently no consensus on the definition of the term, with several authors proposing different definitions. Table [I](#page-2-0) illustrates some of these definitions currently available in the literature.

<span id="page-1-1"></span>![](_page_1_Figure_4.jpeg)

Fig. 1: An example of knowledge graph, where the triplet (v9, r1, v8) serves as an illustration of the link between entities v<sup>9</sup> and v<sup>8</sup> through the relation r<sup>1</sup> and (v8, r2, v1) through r<sup>2</sup> for relation between v<sup>8</sup> and v1.

A KG assumes the form of a directed graph (G), characterized by vertices and edges, where, G = (V, E). Vertices (V ) indicate an entity and Edge (E) between the two vertices conveys the semantic relationship between two entities [\[20\]](#page-16-0). Within the graph, vertices, also referred to as entities or nodes, are linked through relationships represented as edges and facts are depicted using RDF [\[21\]](#page-16-1) triples such as (subject, predicate, object) or (head, relation, tail), denoted by < h, r, t > [\[22\]](#page-16-2). Figure [1](#page-1-1) depicts a simple KG with vertex v<sup>9</sup> and v<sup>8</sup> linked by the relation r1, which goes from v<sup>9</sup> to v8, making a triplet (v9, r1, v8). In this scenario, v<sup>9</sup> represents the head and v<sup>8</sup> represents the tail. In the context of Patient Centric Knowledge Graphs(PCKGs) nodes typically represent entities like patients, drugs, diseases, or genes, whereas edges denote relationships or associations between these entities (see Fig [2\)](#page-2-1).

Over the past few years, especially within the domains of health and biomedicine, various forms of KGs have been proposed, often stemming from literature or Electronic Health Record (EHR), however they are typically not individualcentered. Recent work pertaining to disease-centric knowledge graphs are available in [\[23\]](#page-16-3), [\[24\]](#page-16-4). These knowledge graphs facilitate clinicians' ability to study the relationships between diseases to answer practical clinical problems. Nevertheless, because relationships between disease concepts are frequently distributed across numerous datasets, the concept of integrating multimodal data sources is critical for developing comprehensive disease knowledge graphs [\[23\]](#page-16-3), [\[24\]](#page-16-4). Their robustness of knowledge in terms of diseases and symptoms, on the other hand, must be evaluated in order to assess its accuracy and quality [\[25\]](#page-16-5).

In hospitals, patient information collected during visits, as well as details of test results, clinical notes, symptoms, diagnoses, and drugs, is frequently maintained as an EHR. These records include both unstructured data in the form of free notes, such as progress notes, admission notes, discharge summaries, medical histories, procedures notes, etc., and structured data, such as medical codes, medications, administrative data, vital signs, and laboratory test report data [\[26\]](#page-16-6). Numerous KGs have been introduced in the literature, utilizing EHRs as their foundation. As an illustration, Finlayson SG et al. [\[27\]](#page-16-7) constructed a graph including diseases, drugs, procedures and devices. To do so, they leveraged 1 million clinical concepts from 20 million clinical notes. Similarly, another study in [\[28\]](#page-16-8), formulated a KG covering 156 diseases and 491 symptoms, based on the insights drawn from a dataset of 273,174 patient visits to the emergency department. Although some efforts have been made to incorporate patient-specific information into these graphs, they are not yet patient-centric.

PCKGs can be linked Personal Knowledge Graphs (PKG), an emerging concept in this field. Although the notion of PKGs is relatively new, there have been some notable contributions in this area. One of the pioneering papers that formally introduces the concept of PKG is authored by Balog et al. [\[29\]](#page-16-9). They define PKG *"to be a source of structured knowledge about entities and the relation between them, where the entities and the relations between them are of personal, rather than*<span id="page-2-0"></span>

| Source                | Definition                                                                                                                                                                                                                                                                                                                             |
|-----------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Paulheim et al. [15]  | A knowledge graph (i) mainly describes real world entities and their interrelations, organized in a graph, (ii) defines possible<br>classes and relations of entities in a schema, (iii) allows for potentially interrelating arbitrary entities with each other and (iv)<br>covers various topical domains.                           |
| Ehrlinger et al. [16] | A knowledge graph acquires and integrates information into an ontology and applies a reasoner to derive new knowledge.                                                                                                                                                                                                                 |
| Wang et al. [17]      | A knowledge graph is a multi- relational graph composed of entities and relations which are regarded as nodes and different<br>types of edges, respectively.                                                                                                                                                                           |
| Smirnova et al. [18]  | A Knowledge Graph, also known as a Knowledge Base, is a directed graph formed by triples that connect nodes (subjects) to<br>other nodes (objects) through properties (predicates), where these connections represent semantic relationships and the graph's<br>structure illustrates the subjects, objects, and their interrelations. |
| Duan et al. [19]      | A knowledge graph organizes items, entities, and users as nodes interconnected by edges, providing rich semantics and<br>comprehensive information in a structure akin to natural language.                                                                                                                                            |

TABLE I: Various definition of KGs in available literature.

<span id="page-2-1"></span>![](_page_2_Figure_2.jpeg)

Fig. 2: Illustration of Patient's Clincal Visits Knowledge Graph
*general, importance. The graph has a particular "spiderweb" layout, where every node in the graph is connected to one central node: the user"*. Within the scope of this definition, the PKG only incorporate knowledge pertinent to the specific user.

The emergence of KGs has brought about a substantial transformation in the healthcare domain, providing numerous advantages to this industry. Integrating disparate health data to provide a comprehensive and holistic view of patient health is a significant application of KGs. In the era of precision medicine, wherein therapeutic strategies are customized to suit individual patients' distinctive genetic composition and lifestyle variables, KGs assume a crucial function. Integrating various data types, such as genetic profiles, medical history, lifestyle behaviors, and information from health technology devices, is facilitated to enhance efficiency. Implementing this all-encompassing methodology enables more accurate diagnoses, treatment strategies, and potentially enhanced health results [\[30\]](#page-16-13).

Moreover, KGs are a formidable instrument for deciphering intricate biomedical data, resulting in amplified research findings [\[31\]](#page-16-14). They provide a systematic depiction of connections among different entities, such as genes, diseases, drugs, and pathways, which can be utilized to formulate novel hypotheses and uncover concealed patterns. For example, KGs can elucidate the intricate relationship between a genetic mutation and its association with a specific disease or unveil the influence of environmental factors in the progression of said disease. As mentioned earlier, the discoveries possess the potential to significantly contribute to the development of groundbreaking therapeutic approaches and expand the horizons of biomedical investigation [\[32\]](#page-16-15).

KGs play a crucial role in developing effective decision support systems in the clinical domain. By effectively amalgamating and analyzing individualized patient data, KGs can assist medical practitioners in formulating accurate diagnostic determinations, devising optimal treatment strategies, and forecasting patient prognoses. Consequently, the implementation of KGs can effectively mitigate the occurrence of medical errors [\[33\]](#page-16-16).

KGs have found significant applications across various sectors beyond healthcare. For example, in e-commerce, KGs are extensively used to enhance user experience through personalized recommendations, search result ranking, and semantic search. Prominent e-commerce platforms, such as Walmart, Amazon and eBay, have been utilizing KGs to link items to their respective attributes, thereby providing more relevant product suggestions to customers [\[34\]](#page-16-17). Moreover, KGs are increasingly used in Natural Language Processing (NLP), where they can improve the performance of machine translation, question-answering, and text summarization systems [\[35\]](#page-16-18). KGs also have substantial potential in cybersecurity, as they can aid in mapping relationships between cyber threats, vulnerabilities, and affected systems, facilitating better threat prediction and prevention [\[36\]](#page-16-19). Social networking platforms like Meta and LinkedIn also leverage KGs to understand user relationships and interests, enhancing content recommendation and advertising targeting strategies [\[37\]](#page-16-20). Hence, the applications of KGs are diverse and significantly impact various domains, reinforcing their relevance and potential.

# <span id="page-3-0"></span>III. DEVELOPED TAXONOMY OF PATIENT-CENTRIC KNOWLEDGE GRAPH

Having introduced the concept of KG in the previous section, we now explore the survey's core. This section will examine the various approaches, categorizations, and details that influence PCKGs. PCKGs are characterized by a variety of methodologies and studies. In order to navigate this diverse landscape, we need a well-structured framework. In this context, a taxonomy is a guide that enables readers to comprehend and classify the complexities involved in PCKG construction, evaluation, processing, and applications.

Our developed taxonomy as depicted in Fig [3](#page-4-1) is derived from an exhaustive survey and analysis of existing literature and practices. Based on our findings, we classify PCKG into four major category including *construction, evaluation, process and applications*. Given the heterogeneity of studies in the domain, it provides a structured representation and insight into the why and how of each categorization. The taxonomy presented in this paper outlines essential relationships and highlights the important aspects of each of these categories. Its uniqueness lies in its exhaustive nature, informed by the foundational literature and cutting-edge research. In the following paragraphs, we delve into each of the categories and subcategories.

The first category pertains to *construction*of a PCKG. It is further divided into four sub-categories including*ontology, knowledge sources, knowledge extraction, and knowledge representation.*The initial step,*Ontology*, revolves around designing structured frameworks that are custom-tailored to meet patientcentered needs, ensuring that the graph accurately captures the essence of patient data while also adhering to industry standards and best practices. It is important to articulate distinctions among closely related terms in this context: a Knowledge Graph (KG), which represents interconnected real-world facts; a Knowledge Base (KB), which stores this structured information; and an ontology, which provides a structured framework that illustrates and organizes concepts, ensuring consistent understanding and interpretation of the data within the KG and KB. Second, considering the diverse nature of healthcare data, our *Knowledge Sources*segment highlights the differences and integration methods for three main sources of data: structured, semi-structured, and unstructured data. Third, the*Knowledge Extraction*phase employs techniques like Named Entity Recognition (NER) to pinpoint essential entities such as drugs or symptoms. The process is complemented by Relationship Extraction (RE), which accurately maps the relationships among these entities, giving the graph its characteristic interconnected structure. The last sub-category in construction is*Knowledge Representation*, which focuses on detailing the methodological approach of encapsulating both explicit and implicit knowledge in the PCKG. This involves defining the schema for entities and relationships, establishing constraints, and encoding domain knowledge through axioms and rules, thus constructing a robust framework that effectively translates raw data into a comprehensible and interconnected knowledge graph that accurately mirrors the complexity and depth of patient-centric data.

The second category involves the evaluation of the PCKG, emphasizing the need to assess the efficacy and accuracy of PCKGs post-construction. This assessment can be facilitated by two main methodologies such as *qualitative assessment*and*quantitative assessment*. The former involves a thorough examination of the graph to ensure that it adheres to clinical best practices and contains relevant and correct medical relationships. While the latter makes use of statistical and computational methodologies to evaluate the graph's performance metrics, including accuracy, recall, and precision, and offers a numerical assessment of its ability to accurately and effectively represent and infer knowledge. Once the PCKG has been constructed and evaluated, the next step is its*utilization (processing PCKG)*to derive actionable insights. To attain this objective, various approaches are crucial, including the utilization of methods such as*reasoning, semantic search, and inference*. *Reasoning*enables the PCKG to logically extract new information and knowledge from the existing knowledge within the graph, providing a basis for making intelligent decisions and predictions about healthcare outcomes and strategies. While,*Semantic Search*allows for structured searches to retrieve precise information from the graph.*Inference*, on the other hand, uses the inherent structure of the graph to derive new insights and conclusions.

The final category of the our taxonomy relates to *application*of PCKG in healthcare domain. We explore four primary healthcare use cases, encompassing*recommending individualized interventions, clinical trials, predicting disease before onset, and others.*The first one personalizes patient care treatments based on insights extracted directly from the graph. While the second application aims to enhance the efficiency of*Clinical Trials*, aiding in developing trial designs or optimizing patient selection. On the other hand, a particularly innovative application is *Predicting Disease Before Onset*, where the complex patterns and relationships mapped in the graph can be harnessed to proactively detect potential health issues. Although these are highlighted applications, the potential of

<span id="page-4-1"></span>![](_page_4_Figure_0.jpeg)

Fig. 3: Proposed Taxonomy of Patient-Centric Knowledge Graph

PCKGs extends even further, covering areas such as *drug discovery, personalized medication regimens,*and*preventive health strategies*, to name a few.

Throughout the next sections, we will examine each component of our taxonomy in more detail in order to gain a better understanding of its significance as well as provide an overview of the suggested implementation steps.

# <span id="page-4-0"></span>IV. CONSTRUCTION, EVALUATION AND PROCESSING OF PCKGS

Throughout this section, we explore the complex processes and methodologies involved in developing, assessing, and utilizing PCKG. This comprehensive exploration is structured into four critical categories as illustrated in Fig [3.](#page-4-1) The first category, *construction*of PCKG, explains the complex process of building PCKG, highlighting the significance of ontology, various data sources, knowledge extraction techniques, and knowledge representation. Next,*evaluation*covers both qualitative and quantitative methods for assessing the efficacy and accuracy of these graphs. Similarly, we discuss advanced techniques like*reasoning, semantic search*, and *inference*, which are essential to harnessing the full potential of PCKGs to derive actionable insights and inform healthcare decisions.

## *A. PCKG Construction*The process of creating PCKGs involves four steps. To begin with, the*ontology*phase involves developing structured frameworks based on patient data, which ensure accuracy and adherence to standards.*Knowledge sources*is the second step, which addresses the collection and integration of diverse types of healthcare data, including structured, semi-structured, and unstructured data. During the third step,*knowledge extraction*, key data entities and their interconnections are mapped using named entity recognition and relationship extraction techniques. Lastly, *knowledge epresentation*involves defining a schema for entities and relationships, setting constraints, and encoding domain knowledge to transform raw data into a coherent, interconnected graph reflecting patient data complexity. In the following subsections, we will explore each of these steps in detail.
*1) Ontology:*Ontologies provide a structured framework for representing knowledge, enabling the integration of diverse data sources and facilitating sophisticated access to health information, which is crucial in patient-centered healthcare [\[38\]](#page-16-21). The design and formalization of ontologies involve creating a comprehensive*schema*that integrates information from various*structured*and*unstructured*data sources [\[39\]](#page-16-22). Ontologies are different from knowledge bases and schemas. While an ontology is a formal representation of a set of concepts within a domain and the relationships between those concepts, a knowledge base is a collection of knowledge in a specific area, often built upon an ontology. A schema, on the other hand, is more about the organization of data as a blueprint of how the database is constructed, without delving into the domain-specific relationships and properties that an ontology provides. The application of ontologies in healthcare has been demonstrated in the body of literature. For instance, Sun et al. [\[40\]](#page-16-23) presented an ontology-enabled healthcare service provision model that facilitates joint referral decisions between patients and general practitioners, promoting patient-centric healthcare service provision. The following subsections will expand more on the design and formalization of ontologies.
*a) Design:*The growing digitization of healthcare data and the proliferation of medical information need effective and structured methods of representing and managing patientcentric knowledge. Ontologies as formal representations of knowledge, play a critical role in supporting interoperability, data integration, and decision support [\[41\]](#page-16-24), in healthcare systems. In the context of PCKGs, ontology design is a fundamental process that involves defining and modeling the important entities, connections, and features related to patient health and treatment. Figure [4](#page-5-0) provides an example of a patient's ontology. Authors in [\[42\]](#page-16-25) argue that the rationale behind ontology development is to aid in a shared understanding of information structure among people or software agents, facilitate the reuse of domain knowledge, make domain assumptions explicit, separate domain knowledge from operational knowledge, and analyze domain knowledge. However, there is not standard methodology [\[43\]](#page-16-26) or no single correct way of formulating ontology [\[42\]](#page-16-25).

<span id="page-5-0"></span>![](_page_5_Figure_1.jpeg)

Fig. 4: Illustration of a Basic Patient's Ontology

The initial stage in ontology design is to establish the scope and domain of the PCKG. The scope of the ontology specifies the breadth of medical knowledge included, whereas the domain defines the precise areas of concentration, such as general medical information, specific diseases, or healthcare procedures. Identifying the domain aids in the selection of acceptable current ontologies such as SNOMED-CT [\[44\]](#page-16-27), RxNorm [\[45\]](#page-16-28), and LOINC [\[46\]](#page-16-29) or healthcare standards that can be utilized to maintain interoperability and consistency. While current ontologies and standards may provide a solid foundation, some patient-centric concepts may not be fully covered. In such circumstances, additional classes and features must be added to cater to specific requirements. This requires collaboration between domain experts and ontology engineers to guarantee that the ontological representations are consistent with healthcare practices and guidelines.

The second step is to define the core entities. Entities such as*Patient*, *Disease*, *Medication*,*Symptom*, *Medical Procedure*, *Healthcare Provider*, etc. are typically included. These entities serve as the building blocks for constructing the KG and are crucial for capturing patient-related information.

Next step is to extract relationship or specifying relationship between entities. This step is important for capturing the complex interactions within patient-centric healthcare. Relationships, such as *Diagnosis, Treats, Has Symptom, and Undergoes,*enable meaningful associations and support inferencing capabilities. The developer of ontology during this stage should also carefully consider associated attributes and properties of the entities. For example, attributes of*patient*could be*age*, *gender*, *sex*, *demography*etc. These attributes can influence the diagnosis and treatment decisions [\[47\]](#page-16-30).
*b) Formalization:*Formalizing the ontology focuses on structuring and representing patient data and medical knowledge in a way that is understandable and accessible to healthcare systems. Puustjarvi et al. [\[38\]](#page-16-21) leveraged the Resource Description Framework (RDF) and Web Ontology Language (OWL), to facilitate sophisticated access to health information, a cornerstone for effective patient care. This innovative approach underscores the evolving landscape of healthcare informatics, where the structuring and accessibility of patient data are extremely important.Similarly, Jiang et al. [\[48\]](#page-16-31) provided insight into the role of Formal Concept Analysis (FCA) in the development of ontologies within clinical domains. FCA emerges as a key tool, offering linguistic and contextbased knowledge. This knowledge is indispensable for clinical experts, aiding them in comprehending and applying ontology effectively in their practice. Integrating FCA into ontology development signifies a deeper understanding of clinical data and its nuances, enhancing the overall utility of these ontologies in real-world medical settings.

Further broadening this scope, Djedidi and Aufaure [\[49\]](#page-16-32) presented a methodology that underscores the integration of heterogeneous data sources with existing ontologies and standards. This approach is instrumental in constructing a medical domain ontology as a comprehensive, knowledge-centric decision support system. Such integration is crucial in ensuring the developed ontology is robust and aligns seamlessly with existing medical knowledge frameworks and data sources. Likewise, Mendes, Rodrigues, and Baeta [\[50\]](#page-16-33) made significant contributions to the development of the Ontology for General Clinical Practice (OGCP). This ontology extends the Ontology for General Medical Science (OGMS) by incorporating the Clinical Patient Record (CPR) structure. This extension significantly enhances the representation and reasoning capabilities within clinical practice knowledge, especially in the context of natural language text. The OGCP stands as a testament to the evolving complexity and sophistication required in modern medical ontologies, catering to the nuanced needs of clinical practice.

The dynamic nature of medical knowledge and patient data necessitates methodologies for tracking and updating information. In this context, Ognyanov and Kiryakov [\[51\]](#page-16-34) proposed a formal model for tracking changes in RDF(S) repositories. This model is vital for maintaining the accuracy and relevance of the KG over time, addressing the everchanging landscape of medical data and knowledge. Addressing the user perspective, Clarkson et al. [\[52\]](#page-16-35) introduced a user-centric methodology for the ontology population. This approach aligns user concepts with target ontologies, proving an efficient method for building and maintaining ontologies across various domains. The user-centric approach ensures that the developed ontologies are technically sound and resonate with the needs and understandings of the end-users, be they patients or healthcare professionals.

To bridge the gap between formal ontology and practical application, Ferilli [\[53\]](#page-16-36) proposed an intermediate format that can be easily mapped onto formal ontology for complex reasoning and a graph database for efficient data handling. This innovation represents a significant stride in harmonizing the theoretical aspects of ontology with the practical demands of data management, ensuring that the developed ontologies are both conceptually sound and practically applicable.

Together, these diverse yet interconnected research efforts paint a comprehensive picture of ontology formalization's current state and future potential in patient-centric healthcare. They highlight a collaborative and multi-faceted approach to developing technically robust ontologies that are deeply integrated with the practical realities of healthcare delivery.
*2) Knowledge Sources:*PCKGs are complex structures that integrate diverse data sources, as illustrated in Fig [5,](#page-6-0) in order to offer a comprehensive perspective on a patient's medical past, present ailments, and prospective therapies. The extent of the depth of a PCKG is contingent upon the diversity and quality of the data sources that contribute to its composition. The sources can be classified into three main categories: structured, semi-structured, and unstructured data.

<span id="page-6-0"></span>![](_page_6_Figure_3.jpeg)

Fig. 5: Diverse Sources of Medical and Healthcare Data
*a) Structured Data:*Structured data is organized and easily searchable in relational databases. It follows a specific schema or model, making it straightforward to query and analyze. Structured healthcare data, integral to PCKGs, is exemplified by*EHRs, lab results and dagnostics*, and *genomic and molecular data*. *EHRs*, as digital records of a patient's medical history, encompass diagnoses, medications, treatment plans, and vital statistics, providing a highly structured backbone for PCKGs. Transitioning to *lab results and diagnostics*, this category includes structured, numerical, or categorical data from blood tests, imaging studies (like X-rays, MRIs), and other diagnostic tests, facilitating their integration into PCKGs. Lastly, *genomic and molecular data*, derived from genetic tests, offer structured insights into a patient's disease predisposition, proving essential for the advancement of personalized medicine. Collectively, these structured data forms are key in healthcare analytics and personalized treatment planning.

*b) Semi-Structured Data:* Semi-structured data does not reside in a relational database but does have some organizational properties that make it easier to analyze. It often requires some level of transformation to be fully utilized. Semi-Structured healthcare data encompasses*clinical notes*and*patient-generated data*, each possessing unique characteristics that bridge the gap between structured and unstructured data. *Clinical notes and narratives*, while fundamentally unstructured in text, often adhere to templates or contain tags, rendering them semi-structured and providing contextual depth absent in purely structured datasets. Similarly, *patientgenerated data*, sourced from wearables like fitness trackers and mobile health apps, presents structured data points (e.g., heart rate, steps) but lacks a consistent schema overall, contributing to the semi-structured nature of the data set. This combination of structured elements within an otherwise unstructured framework is valuable for comprehensive healthcare data analysis.

*c) Unstructured Data:*Unstructured data is information that doesn't have a pre-defined data model, making it challenging to analyze using conventional methods. Unstructured healthcare data, characterized by its non-standardized format, includes*patient surveys*and*interviews*as well as*medical literature and research papers*, each offering unique challenges and insights for PCKGs. *Patient surveys and interviews*, typically in text or audio formats, yield qualitative insights into patients' conditions but are unstructured, necessitating advanced analytics for effective integration into PCKGs. Similarly, *medical literature and research papers*are a rich source of valuable insights but remain unstructured; thus, they require NLP techniques to distill and extract relevant information for PCKG incorporation. This unstructured data, with its inherent complexity, plays a crucial role in enhancing the depth and breadth of healthcare analytics and patient care understanding.

By integrating these diverse data sources, PCKGs can offer a multi-dimensional view of a patient's health, thereby enabling more personalized and effective healthcare interventions.
*3) Knowledge Extraction:*In the process of constructing PCKGs, knowledge extraction serves a pivotal role. This stage leverages techniques such as Named Entity Recognition (NER) and Relation Extraction (RE) to distill valuable insights from unstructured data. While NER is focused on identifying and categorizing key entities present in the data, RE takes a step further to identify the relationships that exist between these entities. Through the harmonious interplay of NER and RE, a structured and rich KG can be constructed, paving the way for more personalized and efficient healthcare solutions.
*a) Named Entity Recognition (NER):*NER involves the identification and classification of entities such as diseases, symptoms, and medications from unstructured text. This process is fundamental in transforming raw data into structured, actionable information. Recent advancements in NER methodologies have significantly enhanced the development of KGs, particularly in the medical domain. For instance, Li et al. [\[54\]](#page-16-37) introduced the BiLSTM-Att-CRF model, which integrates an attention mechanism and part-of-speech features to improve clinical NER in Chinese electronic medical records. Building on this, Keretna, Lim, and Creighton [\[55\]](#page-16-38) proposed a graphbased technique that enhances medical NER performance by up to 26% in unstructured medical text, showcasing the versatility of NER applications.

Furthering these developments, Wang and He [\[56\]](#page-16-39) devel-

oped a bi-directional joint embedding model that combines encyclopedic knowledge with original text, thereby showing improved results in Chinese medical NER. Complementing this approach, Wang, Xia, Zhou, Ruan, Gao, and He [\[57\]](#page-16-40) demonstrated that incorporating dictionaries into deep neural networks effectively addresses the challenge of rare and unseen entities in clinical NER, thus broadening the scope of NER applications. In a similar vein, Guan and Tezuka [\[58\]](#page-17-0) highlighted the importance of entity linking and intent recognition in medical question-answering systems, which significantly improves the efficiency and accuracy of medical KG searches. In a similar context, Zhou, Cai, Zhang, Guo, and Yuan [\[59\]](#page-17-1) developed a multi-task adversarial active learning model that enhances medical NER and normalization by considering task-specific features, further illustrating the dynamic evolution of NER methodologies.

These diverse methodologies reflect the evolving landscape of NER in the development of PCKGs. The integration of advanced machine learning techniques, attention mechanisms, and domain-specific knowledge has led to more accurate and efficient entity recognition, which is crucial for the effective utilization of KGs in healthcare.
*b) Relation Extraction (RE):*Similar to NER, RE is a fundamental process in transforming unstructured data into a structured form, enabling more effective data utilization in healthcare and medical research. Recent advancements in RE methodologies, particularly in the context of PCKGs, have been significant. Ruan et al. [\[60\]](#page-17-2) demonstrated that a multi-view graph learning method could notably enhance the precision, recall, and F1-score in relation extraction from Chinese clinical records, outperforming state-of-the-art methods. Similarly, Li et al. [\[61\]](#page-17-3) highlighted the evolution of relation extraction algorithms, emphasizing the role of deep learning, reinforcement learning, active learning, and transfer learning in this domain.

In the realm of therapy-disease KGs, Wang et al. [\[62\]](#page-17-4) constructed a Therapy-Disease KG (TDKG) using entity relationship extraction, achieving an 88.98% accuracy in extracting valid relationships from treatment-disease literature. This underscores the potential of RE in enhancing the accuracy and comprehensiveness of medical KGs. Furthermore, the integration of KGs and lexical features in biomedical relation extraction, as shown by Zhao et al. [\[63\]](#page-17-5), indicates significant improvements in semantic understanding. This approach exemplifies the innovative strategies adopted in recent research to enhance the effectiveness of RE in medical contexts. The development of Conditional Random Field (CRF)-powered classification models with deep learning for clinical relation extraction, as demonstrated by Lv et al. [\[64\]](#page-17-6), further illustrates the ongoing advancements in this field. These models have shown effectiveness in extracting clinical relation information from medical texts, thereby improving the construction of medical KGs.

The field of RE in PCKGs is rapidly evolving, with methodologies leveraging multi-view graph learning, deep learning, and knowledge-enhanced approaches. These advancements are crucial in extracting meaningful and accurate relationships from complex medical texts, thereby enriching the KGs that are essential for advancing patient-centered care and medical research.
*4) Knowledge Representation:*Knowledge Representation (KR) serves as a methodological backbone, enabling the systematic transformation of both explicit and implicit knowledge into a structured and interpretable format. This transformation is crucial for machines to effectively process, analyze, and infer from the data, thereby bridging the gap between raw data and actionable insights.

The process of KR encompasses four key components such as schema definition, constraint establishment, domain knowledge encoding and*axioms and rules utilization*as explained in Section [III.](#page-3-0) Firstly,*schema definition*involves outlining the structure of entities and their relationships within the KG. This step is fundamental in organizing data in a meaningful and accessible manner. Secondly,*constraint establishment*is essential for maintaining data integrity and consistency, ensuring that the relationships and entities adhere to predefined rules and norms. Thirdly, incorporating*domain knowledge*is instrumental in embedding domain-specific intelligence into the system. This allows for a nuanced understanding of the specific area of interest. Fourthly, the integration of*axioms and rules*facilitates advanced reasoning capabilities, enabling the system to infer new knowledge and insights based on the existing data.

These components collectively form a robust framework that simplifies complex data and enhances the system's capability to mirror the intricacies inherent in patient-centric data. In the following subsections, we will delve deeper into each of these components, exploring their roles and significance in the broader context of KR.
*a) Schema Definition:*The concept of*schema definition*in PCKGs revolves around structuring and organizing patient-related data in a way that enhances understanding and interaction with this information. Schemas in KGs serve as blueprints for organizing data, enabling more effective data integration, querying, and analysis. This is particularly crucial in healthcare, where patient data is complex and multifaceted. It aims to create a unified and comprehensive view of patient information, facilitating better healthcare outcomes through informed decision-making. Ghosh and Gilboa [\[65\]](#page-17-7) emphasize the role of memory schemas in coordinating knowledge structures, highlighting the importance of understanding cognitive processes in schema development. This perspective is crucial in PCKGs, where patient data must be organized in a way that aligns with healthcare professionals' cognitive schemas.

Building on the concept of visual schemas by Kranjec et al. [\[66\]](#page-17-8), which enhance comprehension and aid in the diagnostic process, the work of Ohira et al. [\[67\]](#page-17-9) takes a step further by introducing the dynamic nature of graph-based data models. This flexibility is key in the realm of PCKGs, where patient data is not static but continuously evolving. The ability to dynamically add and remove relationships in these KGs ensures that they remain up-to-date and reflective of the latest patient information, a critical aspect in the fast-paced environment of healthcare. Transitioning from the dynamic structuring of patient data, the focus shifts to the cognitive aspects of PCKGs as discussed by Gilboa and Marlatte [\[68\]](#page-17-10). Their research underscores the significance of aligning PCKGs with the mental models of healthcare professionals. This alignment is not just about structuring data efficiently; it's about ensuring that the KG resonates with the users' cognitive processes, thereby enhancing memory recall and decisionmaking capabilities.

Further advancing this discussion, Ji et al. [\[69\]](#page-17-11) delve into the intricacies of schema induction in KGs. Their exploration into the challenges and developments in this field is particularly relevant for PCKGs. As these KGs become more complex and integral to healthcare, understanding and overcoming the challenges in schema induction is paramount for ensuring that PCKGs are robust, comprehensive, and seamlessly integrated into the healthcare workflow.
*b) Constraints Establishment:*The establishment of constraints in PCKGs focuses on enhancing the accuracy, privacy, and efficiency of medical data management and analysis. Various methodologies and strategies adopted in the creation and application of constraints in PCKGs, as evidenced by recent academic research. For example, Aguilar-Escobar et al. [\[70\]](#page-17-12) demonstrate the application of the Theory of Constraints (TOC) in improving the logistics of medical records in hospitals, highlighting the potential of TOC in enhancing service quality and reducing costs in healthcare settings. This approach underscores the importance of efficient data management in PCKGs.

Building on this notion of efficiency, the focus shifts to the critical aspect of privacy and security in healthcare, where Mathew and Obradovic [\[71\]](#page-17-13) illustrate how constraint graphs can secure privacy in medical transactions, preventing unauthorized attribute-based transformations in clinical data. This is crucial in PCKGs, where patient data sensitivity is paramount. The evolution of PCKGs further extends into the domain of dynamic data handling and adaptability. Matsuto and Shiina [\[72\]](#page-17-14) propose a model for medical knowledge representation based on constraints, which dynamically generates laboratory schedules using constraint propagation. This approach emphasizes the adaptability of PCKGs in accommodating patientspecific information.

The adaptability of PCKGs is further enhanced by advancements in KG reasoning. Wu et al. [\[73\]](#page-17-15) discuss a constraint-based embedding model for KG reasoning, focusing on semantic-type constraints in constructing corrupted triplets. This methodology significantly improves the reasoning accuracy of KGs, a key aspect in developing effective PCKGs. Alongside reasoning accuracy, the representation of knowledge and uncertainties also plays an important role. Pugh and Gillan [\[74\]](#page-17-16) introduce Propositional Constraint Graphs (PCG) as a tool for representing knowledge and uncertainties in various tasks, including healthcare. This approach aids in the clear visualization and management of complex information in PCKGs.

The application of these concepts is not limited to static scenarios but also extends to dynamic and evolving medical challenges. Tu [\[75\]](#page-17-17) proposes a constraint propagation approach for identifying biological pathways in COVID-19 KGs. This method, utilizing semantically extracted information, is indicative of the potential for PCKGs in pandemic response and other rapidly evolving medical scenarios. Finally, the integration of context-aware constraints brings a new dimension to the integrity and validation of knowledge in PCKGs. Wilcke et al. [\[76\]](#page-17-18) discuss the use of context-aware constraints to improve knowledge integrity in heterogeneous KGs. This balance of complexity and utility is crucial for knowledge validation tasks in PCKGs [\[76\]](#page-17-18), highlighting the continuous evolution and application of constraints in the development of robust and efficient PCKGs.

The establishment of constraints in PCKGs is a multifaceted process, involving the integration of various methodologies to enhance data accuracy, privacy, and efficiency. The strategies adopted in recent research works reflect a growing emphasis on adaptability, security, and precision in managing patientcentric data in healthcare systems.
*c) Domain Knowledge Encoding:*Domain Knowledge encoding focuses on the integration and representation of complex medical knowledge in a structured and interconnected format. This encoding process is crucial for developing PCKGs, which are instrumental in enhancing patient care through personalized and informed decision-making. The methodologies for creating domain knowledge encoding in PCKGs are diverse and innovative. Wang et al. [\[77\]](#page-17-19) proposed a framework utilizing medical chart and note data, employing a bag-of-words encoding method and a model that considers both global information and local correlations between diseases. This approach underscores the importance of capturing the nuances of medical data for effective KG construction.

Further expanding on these methodologies, Ji et al. [\[78\]](#page-17-20) highlighted the significance of representation space, scoring function, encoding models, and auxiliary information in KG creation. This comprehensive approach indicates the multifaceted nature of KG development, where various components contribute to the robustness and accuracy of the resulting graph. Similarly, Yuan et al. [\[79\]](#page-17-21) presented an approach for biomedical domain KG construction that includes entity recognition, unsupervised entity and relation embedding, latent relation generation, clustering, relation refinement, and relation assignment. This method demonstrates the complexity involved in accurately representing medical knowledge in graph form. Rotmensch et al. [\[28\]](#page-16-8) on the other hand, utilized rudimentary concept extraction and three probabilistic models to construct high-quality health KGs, with the Noisy OR model yielding the best results. This study exemplifies the use of probabilistic models in enhancing the quality of KGs. Furthermore, Yu et al. [\[80\]](#page-17-22) proposed a relationship extraction method for domain KG construction, obtaining upper and lower relationships from structured data, semi-structured data, and unstructured text. This method highlights the importance of extracting relationships from diverse data sources to enrich the KG.

These methodologies contribute to the construction of comprehensive and accurate KGs. In addition, they also pave the way for innovative applications in patient-centric healthcare.
*d) Axioms and Rules Utilization:*PCKGs utilize axioms and rules to enhance the representation and analysis of patient data, leading to more informed healthcare decisions. Axioms in PCKGs are fundamental principles or statements accepted as true without proof, used to define relationships and properties within the graph. Rules, on the other hand, are logical statements that infer new knowledge from existing data within the graph. This combination of axioms and rules plays an important role in structuring and interpreting complex medical data, enabling more personalized and effective patient care.

The methodologies for creating axioms and rules in PCKGs vary across different research works. Chu et al. [\[81\]](#page-17-23) demonstrated the adaptation of multi-center clinical datasets by incorporating an external KG, which enhanced patient features and improved predictions for acute kidney injury in heart failure patients. This approach signifies the importance of integrating diverse data sources and knowledge bases into PCKGs. In another study, Tomczak and Gonczarek [\[82\]](#page-17-24) developed a Graph-based Rules Inducer for extracting decision rules from data streams in diabetes treatment. This method supported medical interviews by tracking hidden context changes and avoiding overfitting, highlighting the importance of dynamic rule adaptation in response to evolving data. Shang et al. [\[83\]](#page-17-25) utilized an EHR-oriented KG system to effectively harness non-used information buried in EHRs, thereby improving healthcare quality and providing interpretable recommendations for specialist physicians.

The use of graph-based association rules for mining medical databases, as demonstrated by Alzoubi [\[84\]](#page-17-26), revealed the potential of discovering hidden medical knowledge, which could contribute significantly to understanding complex medical conditions like preterm birth. Similarly, RuleHub, as introduced by Ahmadi et al. [\[85\]](#page-17-27), is an extensible corpus of rules for public KGs, enabling users to archive and retrieve rules from popular KGs. This tool improves data understanding and reduces redundant work, illustrating the importance of rule sharing and standardization in the field. Wright et al. [\[86\]](#page-17-28) developed a set of rules for inferring patient problems from clinical and billing data, which performed better than using problem list or billing data alone. This method improved clinical decision support and care improvement, showcasing the practical application of rule-based systems in clinical settings.

Based on our review, the utilization of axioms and rules in PCKGs is a dynamic and evolving field, with methodologies ranging from the integration of diverse data sources to the development of specialized, disease-centric graphs. These approaches enhance the understanding of complex medical data and significantly contribute to the advancement of personalized patient care.

#*B. PCKG Evaluation*Evaluation of PCKGs is critical for ensuring their efficacy and accuracy in representing and inferring medical knowledge. These evaluations are typically conducted using two main methodologies:*qualitative*and*quantitative*assessments. Qualitative assessment involves a detailed examination of the KG to ensure it aligns with clinical best practices and accurately reflects medical relationships. This method may usability studies, content analysis, and expert panel reviews to gauge the graph's relevance and correctness [\[28\]](#page-16-8) [\[87\]](#page-17-29). On the other hand, quantitative assessment employs statistical and computational techniques to measure performance metrics such as accuracy, recall, and precision, providing a numerical evaluation of the graph's effectiveness [\[28\]](#page-16-8).
*1) Quantitative Assessment:*Quantitative methodologies for PCKG evaluation focus on numerical metrics to assess the graph's performance. These methods include*completeness, consistency, accuracy, and embedding techniques*.

Completeness in a KG refers to the extent to which all necessary information is represented. In healthcare, this means ensuring that a PCKG includes comprehensive data on diseases, symptoms, treatments, and patient histories. A systematic literature review by Issa et al. [\[88\]](#page-17-30) emphasizes the importance of assessing the completeness of KGs, identifying various methodologies and metrics used for this purpose. On the other hand, consistency involves checking for logical coherence within the graph, particularly in terms of medical terminologies and relationships. This ensures that the KG does not contain contradictory information, which is crucial for clinical decision-making. The work of Sharma and Bhatt [\[89\]](#page-17-31) on privacy-preserving KGs in healthcare highlights the importance of maintaining consistency in data representation.

Other methods like accuracy assessment and embedding techniques are integral to the development of reliable PCKGs. The accuracy of a KG, as highlighted by Li et al. [\[90\]](#page-17-32), is crucial for patient safety and effective treatment planning, ensuring that the information aligns with authoritative medical databases and literature. Complementing this, embedding techniques like node2vec or GraphSAGE, as demonstrated by Talukder et al. [\[91\]](#page-17-33), play a vital role in transforming complex graph data into a more accessible format, thereby enhancing the graph's ability to accurately capture and represent semantic relationships in healthcare applications.

*2) Qualitative Assessment:*Qualitative assessment methodologies are instrumental in evaluating the effectiveness of PCKGs in meeting the needs of their intended users, including clinicians, researchers, and patients.

In the qualitative assessment of PCKGs, usability studies and feedback loops play an important role in the KG evaluation. Usability studies, as highlighted by Brundage et al. [\[87\]](#page-17-29), are crucial for evaluating the ease of use and accessibility of PCKGs, particularly in the context of interpreting patientreported outcomes (PROs) in graphic format. These studies typically employ a blend of qualitative and quantitative methods, including surveys, interviews, and user testing, to gather comprehensive feedback on the user experience. Complementing this, feedback loops, as discussed by Rospocher et al. [\[92\]](#page-17-34), enable users to provide direct input on their experiences with the KG. This feedback is instrumental for the continuous refinement and adaptation of the KG, ensuring it aligns closely with user needs and preferences.

Furthermore, the quality of PCKGs is also gauged through comparison with established databases and expert validation. Benchmarking PCKGs against renowned medical databases like PubMed and ClinicalTrials.gov is essential for assessing their content coverage and accuracy [\[93\]](#page-17-35) [\[94\]](#page-17-36). This comparison helps in identifying any gaps or discrepancies in the PCKG. Additionally, expert validation, underscored by the work of Rotmensch et al. [\[28\]](#page-16-8), involves domain experts who ensure the medical relevance and accuracy of the PCKG's content. Their insights into the clinical applicability of the information within the KG are vital for maintaining its reliability and trustworthiness.

The evaluation of PCKGs through qualitative and quantitative assessments is a multifaceted process that involves both the review of medical content and the analysis of performance metrics. Medical professionals and patients have benefited from progress in this field by ensuring that PCKGs serve as reliable and effective tools.

##*C. PCKG Processing*By processing PCKGs, healthcare providers can extract actionable insights that inform clinical decision. The utilization of PCKGs involve different methods including reasoning, semantic search, and inference.
*1) Reasoning:*KG Reasoning in healthcare extends beyond mere data aggregation, playing an important role in deriving new insights and facilitating informed decision-making. This reasoning process involves structuring information, extracting features and relations, and performing logical deductions to uncover new knowledge from existing data within the graph [\[95\]](#page-17-37). For instance, in the context of mental healthcare, KGs have been utilized for emotion recognition from facial expressions and heart rate, demonstrating the potential of knowledge reasoning in predicting emotional states [\[96\]](#page-17-38). Moreover, KGs in healthcare support clinical decision-making and enhance hospital efficiency by integrating heterogeneous medical knowledge and services, thereby enabling cognitive computing and semantic reasoning [\[97\]](#page-17-39).
*2) Semantic Search:*It enables the retrieval of specific information by structuring searches to navigate the complex relationships within the graph. Various strategies have been adopted to enhance the precision and efficiency of information retrieval in PCKGs. For instance, the use of Semantic Web and Knowledge Management approaches has been pivotal in implementing patient-centric strategies with well-defined semantics [\[98\]](#page-17-40). Graph-based methods that incorporate semanticrich knowledge bases and lazy learning algorithms have shown promise in linking multimodal clinical data for improved diagnosis performance [\[99\]](#page-17-41). Moreover, the retrieval of similar clinical cases has been refined by mapping text to Unified Medical Language System (UMLS) concepts and representing patient records as semantic graphs, demonstrating superiority over traditional models [\[100\]](#page-18-0).
*3) Inference:*Inference uses the graph's inherent structure to deduce new insights, facilitating the discovery of patterns and trends that may not be immediately apparent [\[99\]](#page-17-41) [\[98\]](#page-17-40). These methods underscore the advances in accuracy and efficiency in the processing and utilization of PCKGs, leading to enhanced healthcare outcomes.

After examining the methodologies for building, evaluating, and processing PCKGs, we move on to practical applications and use cases that will demonstrate how PCKGs can be applied in real-world scenarios, demonstrating their impact and significance in healthcare.

### V. APPLICATIONS & USE CASES

<span id="page-10-0"></span>As the healthcare industry increasingly embraces datadriven decision-making, PCKGs have emerged as a powerful tool for personalized medicine. These KGs, which place the patient at the center of a complex network of interconnected health data, offer a holistic view of a patient's health journey. They encompass various data, from medical history and genetic information to lifestyle factors and environmental influences. The applications of these KGs are broad and innovative, offering the potential for improved disease prediction, enhanced treatment planning, and more effective preventive strategies. In the following subsections, we will explore these applications in detail, shedding light on how PCKGs revolutionize healthcare and pave the way for a more personalized, predictive, and proactive approach to patient care.

####*A. Predicting Disease Before Onset*The healthcare sector has seen significant advancements in patient-centric solutions by integrating KGs. Using a KG to proactively forecast the likelihood of a disease developing in an individual before any symptoms appear involves synthesizing various data points related to patient history, genetic information, lifestyle factors, and broader medical knowledge to identify patterns and risk factors associated with diseases. This predictive model allows for early intervention strategies, significantly altering the disease trajectory and improving the patients' long-term health outcomes.

The methodologies for creating PCKGs for disease prediction are diverse and innovative. Chen et al. [\[25\]](#page-16-5) discuss the robust extraction of medical knowledge from EHRs to build graphs that evaluate accuracy across different diseases and patient demographics using non-linear functions for causal relationships. Similarly, Talukder et al. [\[91\]](#page-17-33) describe an AI-based approach integrating disease-related knowledge bodies with Node2VEC for link prediction in disease-symptom networks. Furthermore, Liang et al. [\[101\]](#page-18-1) highlight the use of multihop reasoning over KGs, which provides interpretability and is superior to single-hop methods. In similar vein, the refinement of medical KGs using latent representations aids in prediction and maintains the explainability of diagnoses [\[102\]](#page-18-2). Expanding on these advancements, Li et al. [\[103\]](#page-18-3) introduce a Graph Neural Network-Based Diagnosis Prediction (GNDP) model that uses spatial-temporal graph convolutional networks for diagnosis predictions. In a complementary manner, Zheng et al. [\[104\]](#page-18-4) proposed a Multi-modal Graph Learning framework (MMGL) that exploits multi-modality information for disease prediction.

Different strategies across research works include the use of random walk along KG [\[105\]](#page-18-5), Graph Neural Networks (GNN) for embedding medical concepts [\[106\]](#page-18-6), and hybrid systems combining KGs with clinical experience for pediatric disease prediction [\[107\]](#page-18-7). Zhang et. al [\[108\]](#page-18-8) built an automatic question-answering system based on medical KGs, while Saha et al. [\[109\]](#page-18-9) predicted missing and noisy links in clinical KGs using neighborhood-based embeddings. Key findings from our literature review indicate significant advances in disease prediction accuracy, efficiency, and outcomes. For instance, the integration of GNNs with EMR data has led to highly representative node embeddings that improve prediction accuracy [\[106\]](#page-18-6) . The use of tensor factorization on biological KGs for predicting co-morbid disease pairs [\[110\]](#page-18-10) and the construction of KGs based on evidence-based medicine for diabetes complications [\[111\]](#page-18-11) are notable advances.

PCKGs represent a significant step forward in the field of disease prediction. They offer the potential to transform patient outcomes by enabling early detection and personalized treatment plans. Future implications include the continued refinement of these models to enhance their predictive power and the integration of even more diverse data sources to capture the full spectrum of patient health and disease progression. In Table [II](#page-12-0) we summarize a list of selected disease prediction applications in terms of their impact and limitations in the field of PCKGs.

##*B. Recommending Individualized Interventions*Intervention recommendation aims to use KGs to suggest tailored medications and treatments. By mapping patients' unique health profiles, these graphs enable physicians to pinpoint optimal therapies that enhance the precision of medical decisions, leading to improved patient outcomes. The application of KGs in personalized treatments is predicated on the integration of diverse data sources, including EHRs, clinical notes, and patient-generated data, to construct a comprehensive, interconnected data structure that reflects individual patient profiles. This patient-centric approach is crucial as it allows for treatments to be tailored based on the unique medical history, genetic information, lifestyle, and preferences of each patient, which is a departure from the one-size-fits-all healthcare model.

The approaches to developing and implementing PCKGs differ among research studies, but they all aim to achieve robustness, accuracy, and personalization. For instance, Gyrard et al. [\[113\]](#page-18-12) aggregate knowledge from IoT devices, clinical notes, and EMRs to manage chronic diseases, showcasing the integration of AI and machine learning in constructing Personalized Healthcare Knowledge Graphs (PHKGs). Transitioning from a general approach to a more specialized application, Individualized Knowledge Graphs (IKGs) in cardiovascular medicine are one strategy for developing these KGs, which combine biological knowledge with medical histories and health outcomes to create personalized treatment strategies [\[114\]](#page-18-13). Further refining the methodological framework, Rotmensch et al. [\[28\]](#page-16-8) utilized concept extraction and probabilistic models, finding the Noisy OR model particularly effective for constructing high-quality health KGs. Building on these methodologies, Shirai, et al. [\[4\]](#page-15-3) applied Personal Knowledge Graphs (PKGs) to integrate patient-specific information into decision-making tools for personalized healthcare. In a similar vein of enhancing disease treatment strategies through personalized data, Zhu et al. [\[115\]](#page-18-14) introduced a KG that enhances rare disease (RD) treatment recommendations by systematically compiling and semantically annotating RD-related scientific articles, aggregating essential research findings and therapeutic insights with a sophisticated data model.

Literature has demonstrated significant advances in treatment accuracy, efficiency, and outcomes. For instance, the use of Four-Tuple Path Matrix in Traditional Chinese Medicine has been proposed to create personalized KGs, enhancing diagnostic modalities [\[116\]](#page-18-15). Li et al. [\[117\]](#page-18-16) demonstrated how personal KGs could be automatically constructed from user utterances in conversational dialogs, indicating the potential for realtime, dynamic treatment adjustments. PCKGs provide multiple advantages. By analyzing patient profiles using KGs, treatments are more accurate and efficient than traditional methods. For example, Zhang et al. [\[118\]](#page-18-17) developed an intuitive graph representation of knowledge for nonpharmacological treatment of psychotic symptoms in dementia, which could potentially transform care strategies for such complex conditions.

There is a wide range of potential implications for future personalized treatment and patient outcomes. A more informed and dynamic approach to treatment has the potential to enhance medical precision, improve patient engagement, and optimize health outcomes by integrating PCKGs. It is expected that these methods will become more sophisticated as the field evolves, allowing for even greater personalization and efficacy in treatment. In Table [III](#page-13-1) we summarize a list of selected treatment decision applications in terms of their impact and limitations in the field of PCKGs.

###*C. Enhancing Clinical Trials*KGs represent a paradigm shift in clinical trail patient selection by offering a structured, interconnected data framework that can encapsulate complex patient information, medical histories, and potential trial criteria. In clinical trials, patientcentric approaches are crucial because personalized medicine tailors treatments to each patient's characteristics, necessitating a comprehensive understanding of patient data.

PCKGs are created and applied using diverse methodologies. Gortzis and Nikiforidis [\[119\]](#page-18-18) described an N-tier system that combines KGs with human collaboration and scalable knowledge engineering tactics. In order to select patients effectively, expert input must be combined with scalable data

<span id="page-12-0"></span>

| Paper Title                                                                                                                 | Focus/Objective                                                                                                                                                                                                                                          | Contribution(s)                                                                                                                                                                                                                                                                                               | Limitation(s)                                                                                                                                                                                                                               |
|-----------------------------------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
|                                                                                                                             |                                                                                                                                                                                                                                                          | Predicting Disease Before Onset                                                                                                                                                                                                                                                                               |                                                                                                                                                                                                                                             |
| A novel link prediction approach on<br>clinical knowledge graphs utilizing<br>graph structures [112]                        | The goal is to create a proof of<br>concept showcasing the efficacy of<br>graph structures in AI methodolo<br>gies.                                                                                                                                      | • Development of a graph-based<br>method<br>merging<br>Conditional<br>Random<br>Fields<br>(CRFs)<br>and<br>graph embedding for knowledge<br>discovery.<br>• This method successfully pre<br>dicts labels for graph nodes with<br>high precision and recall.                                                   | • The<br>method<br>is<br>time-intensive<br>when querying features from the<br>graph,<br>particularly<br>with<br>large<br>datasets.<br>• Increased runtime and memory<br>demands for multi-node paths in<br>the graph.                       |
| Deep knowledge reasoning guided<br>disease prediction [101]                                                                 | The paper aims to enhance model<br>interpretability<br>by<br>integrating<br>knowledge entities with single-hop<br>and multi-hop relationships                                                                                                            | • Introduction of HitaNet, a hier<br>archical time-aware attention net<br>work.<br>• HitaNet<br>uses<br>a<br>self-attention<br>based transformer model for en<br>hanced disease prediction.<br>• The paper presents a unique to<br>ken wrapping method to merge<br>knowledge graph insights with<br>EHR data. | • Lower time efficiency compared<br>to baseline methods as a limita<br>tion.<br>• The proposed method's effective<br>ness diminishes with the avail<br>ability of sufficient data.                                                          |
| Predicting missing and noisy links<br>via neighbourhood preserving graph<br>embeddings in a clinical knowlege<br>base [109] | The<br>paper<br>proposes<br>a<br>model<br>that<br>combines<br>support<br>vector<br>classification<br>(SVC)<br>and<br>neural<br>network-based<br>probabilistic<br>embedding<br>(NPE)<br>to<br>predict<br>the links between clinical entities<br>in the KG | • The development of a novel ap<br>proach that integrates SVC and<br>NPE for link prediction in KGs.<br>• The model achieved promising<br>results in terms of predicting<br>missing associations.                                                                                                             | • The reliance on existing clini<br>cal databases, which may contain<br>noisy or incomplete data.<br>• The potential mismatch between<br>model predictions and clinical<br>recommendations.                                                 |
| Relation prediction of co-morbid<br>diseases<br>using<br>knowledge<br>graph<br>completion [110]                             | The objective is to predict relation<br>ships between co-morbid diseases<br>using knowledge graph completion<br>techniques                                                                                                                               | • Proposing a method that com<br>bines distributed representation<br>learning and graph embedding to<br>predict disease-disease relation<br>ships<br>• Development of a novel approach<br>that leverages the structure and<br>semantics of a KG to predict dis<br>ease relationships                          | • The proposed method relies on<br>the<br>availability<br>of<br>a<br>compre<br>hensive and accurate knowledge<br>graph.<br>• The evaluation of the method is<br>limited to a specific dataset.                                              |
| Disease prediction via graph neural<br>networks [106]                                                                       | The paper addresses the challenges<br>in predicting both common and<br>rare diseases by integrating expert<br>knowledge with machine learning<br>techniques.                                                                                             | • Proposing a systematic solution<br>that combines expert knowledge<br>with machine learning.<br>• Introducing<br>a<br>novel<br>graph<br>embedding-based<br>model<br>for<br>disease prediction, which learns<br>embeddings<br>from<br>medical<br>concept<br>graphs<br>and<br>patient<br>record graphs.        | • Struggling to adapt to rare dis<br>eases due to data scarcity and<br>complex symptom-diagnosis re<br>lationships.<br>• Reliance<br>on<br>historical<br>patient<br>records for model training limits<br>the ability to serve new patients. |

TABLE II: A summary of selected literature on using PCKGs for*"Predicting Disease Before Onset"*structures. Xiang et al. [\[120\]](#page-18-20) highlighted the standardization and structural integration provided by KGs, which are essential for auxiliary diagnosis systems in clinical trials. Strategies for developing KGs for patient selection in clinical trials include the linkage of multimodal data types for automatic diagnosis [\[99\]](#page-17-41). Nicholson and Greene [\[121\]](#page-18-21) discussed machine learning methods for constructing low-dimensional representations of KGs, which support applications in genomic, pharmaceutical, and clinical domains.

Various studies have indicated significant improvements in the accuracy, efficiency, and outcomes of clinical trial patient selection. For instance, the Safe Medicine Recommendation (SMR) framework by Wang et al. [\[122\]](#page-18-22) bridges electronic medical records with medical KGs to learn patient-diseasedrug embeddings, enhancing the precision of clinical trial patient selection. Compared with traditional ways of selecting patients for clinical trails, PCKGs offer a more thorough and subtle approach. Several studies have demonstrated the effectiveness of KGs in improving clinical trial design and outcomes by providing a more holistic view of patient data, facilitating personalized trial matching, and providing a more holistic view of patient data [\[123\]](#page-18-23) [\[124\]](#page-18-24) [\[93\]](#page-17-35). Utlimately, the integration of patient-centric KGs in the selection of clinical trial participants has the potential to transform the field by improving the precision and personalization of patient care. As a result of a better understanding of patient data, future trials will likely be more adaptive, patients will be more engaged, and outcomes may be improved. In Table [IV](#page-14-0) we summarize a list of selected treatment decision applications in terms of their impact and limitations in the field of PCKGs.

Although PCKGs have proven their value in areas such as recommending individual interventions, predicting disease

<span id="page-13-1"></span>

| Paper Title                                                                                                                                   | Focus/Objective                                                                                                                                                                                                                        | Contribution(s)                                                                                                                                                                                                                                             | Limitation(s)                                                                                                                                                                                                                                                                                                |
|-----------------------------------------------------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
|                                                                                                                                               |                                                                                                                                                                                                                                        | Recommending Individualized Interventions                                                                                                                                                                                                                   |                                                                                                                                                                                                                                                                                                              |
| Individualized<br>Knowledge<br>Graph<br>[114]                                                                                                 | • Envisioning<br>individualized<br>Knowledge<br>Graphs<br>(iKGs)<br>in<br>cardiovascular medicine.<br>• Proposing a modern informatics<br>platform for transforming clinical<br>and scientific discovery.                              | • Introducing the concept of iKGs<br>for<br>aggregating<br>and<br>present<br>ing individualized cardiovascular<br>health data.<br>• Highlighting the role of iKGs<br>in linking biological and clinical<br>knowledge of individual patients.                | • Acknowledging<br>challenges<br>in<br>data<br>fragmentation,<br>noncommensurability,<br>and<br>semantic<br>inference<br>within<br>cardiovascular data.                                                                                                                                                      |
| Personalized<br>Health<br>Knowledge<br>Graph [113]                                                                                            | Aims to manage chronic diseases<br>more effectively using IoT data an<br>alytics and explicit knowledge.                                                                                                                               | • Proposes<br>a<br>methodology<br>to<br>build<br>PHKG,<br>integrating<br>heterogeneous data sources.<br>• Offers a solution for contextu<br>alizing and personalizing health<br>care information.                                                           | • The<br>paper<br>acknowledges<br>the<br>complexity in semantic integra<br>tion of diverse data.<br>• It highlights the challenges in tai<br>loring generic knowledge to indi<br>vidual patients.                                                                                                            |
| Developing an intuitive graph repre<br>sentation of knowledge for nonphar<br>macological treatment of psychotic<br>symptoms in dementia [118] | • Develop a knowledge graph for<br>nonpharmacological treatment of<br>psychotic symptoms in dementia.<br>• Enhance understanding and man<br>agement of dementia-related psy<br>chotic symptoms through non<br>pharmacological methods. | • Creation<br>of<br>the<br>Dementia<br>Related<br>Psychotic<br>Symptom<br>Nonpharmacological<br>Treatment<br>Ontology (DRPSNPTO).<br>• Improvement in visualization and<br>computerization of gerontologi<br>cal knowledge.                                 | N/A                                                                                                                                                                                                                                                                                                          |
| Learning a health knowledge graph<br>from<br>electronic<br>medical<br>records<br>[28]                                                         | Automatically learn a health KG<br>from EMRs to link diseases and<br>symptoms<br>and<br>improve<br>clinical<br>decision-support systems.                                                                                               | • A<br>methodology<br>for<br>deriving<br>health<br>KG<br>from<br>EMR<br>using<br>probabilistic models.<br>• Demonstration that the noisy OR<br>model significantly outperforms<br>other tested models.                                                      | • Inherent difficulties in interpret<br>ing EMR data, especially the<br>presence of complex patient con<br>ditions.<br>• The reliance on rudimentary con<br>cept extraction pipelines.<br>• Limitations related to the au<br>tomatic inference of causal re<br>lationships<br>from<br>observational<br>data. |
| Applying<br>personal<br>knowledge<br>graphs to health [4]                                                                                     | The<br>paper<br>focuses<br>on<br>leverag<br>ing<br>PHKGs<br>to<br>enhance<br>health<br>care decision-making by integrat<br>ing personal health information with<br>broader knowledge graphs.                                           | • Proposing a conceptual frame<br>work for PHKGs, highlighting<br>how they can support person<br>alized, knowledge-driven health<br>care<br>applications<br>by<br>leverag<br>ing data from EHRs, IoT de<br>vices, and other health-related<br>data sources. | • Collecting and storing personal<br>health knowledge from heteroge<br>neous sources.<br>• Linking personal health knowl<br>edge to external KGs enhances<br>the PHKG with broader contex<br>tual information.<br>• Maintaining the PHKG to ensure<br>it remains up-to-date and accu<br>rate.                |

TABLE III: A summary of selected literature on using PCKGs for*"Recommending Individualized Intervention"*before onset, and improving clinical trials, their utility goes far beyond these. It is increasingly becoming common to use KGs in other innovative healthcare applications, such as optimizing hospital workflows, tailoring patient engagement strategies, and even developing telemedicine platforms. With these innovative tools, healthcare can be revolutionized by providing a more holistic, integrated view of patient data, as well as new opportunities for research and treatment methods.

Having explored the diverse applications and use cases of PCKGs in various domains, we now turn our attention to the challenges and future directions in this field. This transition allows us to critically examine the current limitations and envision potential advancements that could further enhance the utility and effectiveness of PCKGs.

#### VI. RESEARCH CHALLENGES & DISCUSSION

<span id="page-13-0"></span>PCKGs in healthcare are designed to provide a comprehensive, unified view of patient data by integrating information from various sources, including EHRs, medical literature, and patient-generated data. These graphs aim to support better clinical decision-making and personalized patient care by representing complex medical data in an interconnected format that is more accessible and actionable for healthcare providers.

The current state of research in PCKGs is focused on overcoming several key challenges to maximize their potential in healthcare. Ji et al. [\[78\]](#page-17-20) discuss the difficulties in knowledge acquisition, completion, and temporal KG development, which are crucial for maintaining up-to-date and comprehensive patient profiles. Building on this foundation, Chen et al. [\[25\]](#page-16-5) highlight the need for robustness in PCKGs, particularly in addressing sample size limitations and unmeasured confounders, to extend models to larger patient visits. Moreover, Rastogi and Zaki [\[125\]](#page-18-25) emphasize the importance of designing, building, and operationalizing PCKGs that are tailored to individual patients. However, a significant hurdle remains in the actual construction of PCKGs. As noted by Gyrard et al. [\[113\]](#page-18-12) and

<span id="page-14-0"></span>

| Paper Title                                                                                    | Focus/Objective                                                                                                                                                                                      | Contribution(s)                                                                                                                                                                                                                                                                                                                                         | Limitation(s)                                                                                                                                                                                                                                                                                          |  |  |  |  |
|------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--|--|--|--|
| Recommending Individualized Interventions                                                      |                                                                                                                                                                                                      |                                                                                                                                                                                                                                                                                                                                                         |                                                                                                                                                                                                                                                                                                        |  |  |  |  |
| Knowledge graph-based clinical de<br>cision<br>support<br>system<br>reasoning<br>[120]         | The focus of the paper is to high<br>light the benefits of using knowl<br>edge graphs over traditional hand<br>crafted rule databases in CDSSs                                                       | The<br>introduction<br>of<br>the<br>Path<br>Ranking<br>Algorithm<br>(PRA)<br>as<br>a<br>method for automatically discover<br>ing symptoms without human inter<br>vention.                                                                                                                                                                               | • The probability of certain paths<br>in the KG may not be accurate.<br>• The lack of details on the classi<br>fication model used.                                                                                                                                                                    |  |  |  |  |
| Constructing knowledge graphs and<br>their biomedical applications [121]                       | • Examining the construction and<br>application of biomedical knowl<br>edge graphs.<br>• Emphasizing how machine learn<br>ing is transforming these pro<br>cesses.                                   | • Discussion of knowledge graph<br>construction,<br>including<br>manual<br>curation and text mining.<br>• Review of representational learn<br>ing techniques and their applica<br>tions in biomedical fields.                                                                                                                                           | • Need for advanced techniques to<br>handle complex sentence struc<br>tures.<br>• Limitations in current methods to<br>represent diverse relationships in<br>KGs.<br>• Scalability and memory limita<br>tions in matrix factorization tech<br>niques.                                                  |  |  |  |  |
| SMR: Medical knowledge graph<br>embedding for safe medicine rec<br>ommendation [122]           | Developing a framework to recom<br>mend safe medicines by leverag<br>ing a heterogeneous graph that in<br>tegrates patient data, diseases, and<br>medicines.                                         | • Development of graph-based em<br>bedding models enabling the rec<br>ommendation of newly emerged<br>medicines effectively.<br>• A novel method to recommend<br>safe medicines for new patients<br>and minimizing potential adverse<br>drug reactions.<br>• Introduction of the SMR frame<br>work as a new approach to the<br>link prediction problem. | • Dealing with the challenge of<br>recommending<br>safe<br>medicines,<br>especially new ones, to patients.<br>• Minimizing<br>potential<br>adverse<br>drug<br>reactions<br>in<br>medicine<br>recommendations is critical to<br>patient safety.                                                         |  |  |  |  |
| Patient centric approach for clinical<br>trials: Current trend and new oppor<br>tunities [123] | Exploring<br>the<br>shifting<br>paradigm<br>in clinical trials towards a more<br>patient-centric model.                                                                                              | Identifying new opportunities for<br>the clinical research industry to<br>adopt patient-centric approaches to<br>accelerate drug development and<br>improve trial outcomes.                                                                                                                                                                             | • The complexity and rising costs<br>of clinical research.<br>• Ensuring data transparency and<br>building trust with patients par<br>ticipating in clinical trials.                                                                                                                                   |  |  |  |  |
| Automatic diagnosis with efficient<br>medical case searching based on<br>evolving graphs [99]  | Developing<br>a<br>method<br>for<br>auto<br>matic diagnosis by improving med<br>ical case searching using evolving<br>graphs, which dynamically incorpo<br>rate new medical cases and knowl<br>edge. | • Introduction of an evolving graph<br>framework that integrates new<br>medical cases and knowledge.<br>• A<br>novel<br>method<br>for<br>medical<br>case searching that leverages the<br>evolving graph structure.<br>• An optimization strategy for em<br>bedding learning in the heteroge<br>neous graph.                                             | • Handling the dynamic nature of<br>medical knowledge and cases.<br>• Balancing<br>the<br>computational<br>complexity<br>of<br>embedding<br>learning<br>in<br>a<br>continuously<br>evolving graph structure.<br>• Scalability and maintaining high<br>accuracy and efficiency as the<br>graph expands. |  |  |  |  |

Cong et al. [\[126\]](#page-18-26) point out the complexities in constructing PCKGs, which are often time-consuming and heavily reliant on the quality of source data. A significant challenge in the application of traditional KG embedding methods to patientcentric healthcare applications is their struggle with structural sparsity. Hu et al. [\[127\]](#page-18-27) argue that conventional methods, such as TransE and ConvE, while adept at mapping entities and relationships into a vector space, falter due to their reliance solely on KG triplets, neglecting the rich auxiliary texts that describe entities. This limitation significantly limits the comprehensiveness and utility of KGs in capturing detailed patient information and medical knowledge, indicating a key challenge in leveraging KGs for complex healthcare applications.

Data quality and standardization remain significant challenges, as heterogeneous data structures, poor data quality, and varying medical standards complicate the integration of data into a coherent KG Zhang et al. [\[128\]](#page-18-28). The integration of PCKGs into clinical workflows also presents challenges, as it requires the development of systems that complement healthcare providers' routines without causing disruptions [\[119\]](#page-18-18). Additionally, scalability is another concern, as PKGs must be able to incorporate an ever-increasing amount of data from diverse sources, including genomic information and patient lifestyle data [\[99\]](#page-17-41). Patient data privacy is a critical concern, particularly in the context of utilizing PCKGs, which involve handling sensitive personal health information and require strict privacy controls to safeguard patient confidentiality. The use of patients' data for various purposes, such as consultations, research, and emergency, poses a significant challenge for authorization systems, emphasizing the need for robust privacy protection [\[129\]](#page-18-29) [\[89\]](#page-17-31). Furthermore, real-time data analysis within PCKGs is technologically demanding, requiring advanced computational methods to process and analyze data promptly for it to be clinically relevant [\[130\]](#page-18-30).

In light of these challenges, it becomes clear that continued technological and methodological innovations are necessary to enhance the predictive capabilities of PCKGs. The use of advanced analytics, machine learning, and semantic web technologies could be key. Additionally, as PCKGs become more integrated into healthcare delivery, it becomes increasingly important to address regulatory and ethical considerations. This requires collaboration among computer scientists, healthcare professionals, and policymakers to align the development of PCKGs with broader healthcare objectives.

#### VII. CONCLUSION & FUTURE DIRECTIONS

<span id="page-15-10"></span>This literature review of PCKGs explores their development, evaluation, processing techniques, applications, challenges, and prospects. PCKGs represent a field in healthcare informatics that aims to revolutionize personalized patient care by integrating and synthesizing diverse healthcare data sources. The review highlights the current state-of-the-art methodologies for constructing and evaluating PCKGs, emphasizing the importance of qualitative and quantitative approaches to assess their effectiveness in healthcare settings. In addition to construction and evaluation, the review delves into innovative processing techniques such as reasoning, semantic search, and inference. These techniques significantly enhance the accuracy and efficiency of PCKGs, ultimately improving patientcentered care. Furthermore, exploring different applications of PCKGs in healthcare—including disease prediction, personalized treatment recommendations, and advancements in clinical trials—reveals their potential to transform healthcare through personalized and predictive medicine.

Future directions in PCKG research include leveraging advanced analytics and machine learning to improve predictive capabilities, which could lead to more accurate and timely interventions [\[131\]](#page-18-31). In addition, semantic web technologies are also predicted to play a significant role in enhancing the accessibility and utility of PCKGs [\[93\]](#page-17-35). Building on this momentum, personalized medicine emerges as a promising area where PCKGs can make a substantial impact. By linking genomic data with clinical outcomes, treatments can be tailored to individual patients, offering a more personalized approach to healthcare [\[132\]](#page-18-32). To further this advancement, methodological innovations, such as new algorithms for data harmonization and user interface design, are crucial. These innovations are needed to address current challenges and facilitate the broader adoption of PCKGs in clinical practice [\[133\]](#page-18-33). Simultaneously, as PCKGs become more integrated into healthcare delivery, regulatory and ethical considerations gain prominence. These aspects are critical to ensuring that the deployment of PCKGs adheres to the highest standards of patient care and data management. Therefore, cross-disciplinary collaboration becomes essential for advancing PCKG technology. This involves computer scientists, healthcare professionals, and policy-makers working together to ensure that the development of PCKGs aligns with broader healthcare objectives and respects ethical guidlines [\[134\]](#page-18-34) [\[135\]](#page-18-35).

PCKGs represent a significant advancement in healthcare informatics. They can transform patient outcomes by enabling early detection and personalized treatment plans. Despite the challenges involved, the future outlook for PCKGs is promising as they have the potential to improve patient care and healthcare delivery significantly. Ongoing research efforts and interdisciplinary collaboration will be crucial in fully realizing their novel impact on healthcare.

#### ACKNOWLEDGMENTS

This work was supported by PATENT Lab (Predictive Analytics and TEchnology iNTegration Laboratory) at the Department of Computer Science and Engineering, Mississippi State University.

#### CONFLICT OF INTEREST STATEMENT

The authors declare that the research was conducted in the absence of any commercial or financial relationships that could be construed as a potential conflict of interest.

#### AUTHOR CONTRIBUTIONS

All authors contributed to producing the paper. All authors contributed to the article and approved the submitted version.

#### REFERENCES

- <span id="page-15-0"></span>[1] B. Mesko and Dave deBronkart. Patient design: The importance of ´ including patients in designing health care.*Journal of Medical Internet Research*, 24, 2022.
- <span id="page-15-1"></span>[2] B. Blobel. Educational challenge of health information systems' interoperability. *Methods of Information in Medicine*, 46:52–56, 2007.
- <span id="page-15-2"></span>[3] B. Blobel. Ontology driven health information systems architectures enable phealth for empowered patients. *International journal of medical informatics*, 80(2):e17–e25, 2011.
- <span id="page-15-3"></span>[4] S. Shirai, O. Seneviratne, and D. McGuinness. Applying personal knowledge graphs to health. *ArXiv*, abs/2104.07587, 2021.
- <span id="page-15-4"></span>[5] Nabila Albannai, Yasemin Gultepe, and Abdulmawla M. A. Najih. A review of machine learning applications in semantic web. *International Journal of Scientific and Technological Research*, 2019.
- <span id="page-15-5"></span>[6] M. Almunawar and M. Anshari. Improving customer service in healthcare with crm 2.0. *ArXiv*, abs/1204.3685, 2012.
- <span id="page-15-6"></span>[7] Finlay MacLean. Knowledge graphs and their applications in drug discovery. *Expert Opinion on Drug Discovery*, 16:1057 – 1069, 2021.
- <span id="page-15-7"></span>[8] V. Alagar, K. Periyasamy, and Kaiyu Wan. Privacy and security for patient-centric elderly health care. In *2017 IEEE 19th International Conference on e-Health Networking, Applications and Services (Healthcom)*, pages 1–6, 2017.
- <span id="page-15-8"></span>[9] L. Harper and M. Honour. Working together to personalize "the patient". *Newborn and Infant Nursing Reviews*, 15:21–23, 2015.
- <span id="page-15-9"></span>[10] K. Goniewicz, E. Carlstrom, A. Hertelendy, F. Burkle, M. Goniewicz, ¨ D. Lasota, J. Richmond, and A. Khorram-Manesh. Integrated healthcare and the dilemma of public health emergencies. *Sustainability*, 13:4517, 2021.
- <span id="page-15-11"></span>[11] Richard H Richens. Preprogramming for mechanical translation. *Mech. Transl. Comput. Linguistics*, 3(1):20–25, 1956.
- <span id="page-15-12"></span>[12] Allen Newell, John C Shaw, and Herbert A Simon. Report on a general problem solving program. In *IFIP congress*, volume 256, page 64. Pittsburgh, PA, 1959.
- <span id="page-15-13"></span>[13] Edward Shortliffe. *Computer-based medical consultations: MYCIN*, volume 2. Elsevier, 2012.
- <span id="page-15-14"></span>[14] Xin Dong, Evgeniy Gabrilovich, Geremy Heitz, Wilko Horn, Ni Lao, Kevin Murphy, Thomas Strohmann, Shaohua Sun, and Wei Zhang. Knowledge vault: A web-scale approach to probabilistic knowledge fusion. In *Proceedings of the 20th ACM SIGKDD international conference on Knowledge discovery and data mining*, pages 601–610, 2014.
- <span id="page-15-15"></span>[15] Heiko Paulheim. Knowledge graph refinement: A survey of approaches and evaluation methods. *Semantic web*, 8(3):489–508, 2017.
- <span id="page-15-16"></span>[16] Lisa Ehrlinger and Wolfram Woß. Towards a definition of knowledge ¨ graphs. *SEMANTiCS (Posters, Demos, SuCCESS)*, 48(1-4):2, 2016.

- <span id="page-16-10"></span>[17] Quan Wang, Zhendong Mao, Bin Wang, and Li Guo. Knowledge graph embedding: A survey of approaches and applications. *IEEE Transactions on Knowledge and Data Engineering*, 29(12):2724–2743, 2017.
- <span id="page-16-11"></span>[18] A. Smirnova, J. Audiffren, and P. Cudre-Mauroux. Distant supervision ´ from knowledge graphs. In *Encyclopedia of Big Data Technologies*, pages 1–7, 2018.
- <span id="page-16-12"></span>[19] Y. Duan, L. Shao, G. Hu, Z. Zhou, Q. Zou, and Z. Lin. Specifying architecture of knowledge graph with data graph, information graph, knowledge graph and wisdom graph. In *2017 IEEE 15th International Conference on Software Engineering Research, Management and Applications (SERA)*, pages 327–332, 2017.
- <span id="page-16-0"></span>[20] Antoine Bordes, Jason Weston, Ronan Collobert, and Yoshua Bengio. Learning structured embeddings of knowledge bases. In *Proceedings of the AAAI conference on artificial intelligence*, volume 25, pages 301–306, 2011.
- <span id="page-16-1"></span>[21] Richard Cyganiak, David Wood, and Markus Lanthaler. Rdf 1.1 concepts and abstract syntax. World Wide Web Consortium (W3C), February 2014. Accessed: 2023-08-29.
- <span id="page-16-2"></span>[22] Bilal Abu-Salih. Domain-specific knowledge graphs: A survey. *Journal of Network and Computer Applications*, 185:103076, 2021.
- <span id="page-16-3"></span>[23] Payal Chandak, Kexin Huang, and Marinka Zitnik. Building a knowledge graph to enable precision medicine. *Scientific Data*, 10(1):67, 2023.
- <span id="page-16-4"></span>[24] Yucong Lin, Keming Lu, Sheng Yu, Tianxi Cai, and Marinka Zitnik. Multimodal learning on graphs for disease relation extraction. *Journal of Biomedical Informatics*, page 104415, 2023.
- <span id="page-16-5"></span>[25] I. Chen, M. Agrawal, S. Horng, and D. Sontag. Robustly extracting medical knowledge from ehrs: A case study of learning a health knowledge graph. In *Pacific Symposium on Biocomputing. Pacific Symposium on Biocomputing*, volume 25, pages 19–30, 2019.
- <span id="page-16-6"></span>[26] Tabinda Sarwar, Sattar Seifollahi, Jeffrey Chan, Xiuzhen Zhang, Vural Aksakalli, Irene Hudson, Karin Verspoor, and Lawrence Cavedon. The secondary use of electronic health records for data mining: Data characteristics and challenges. *ACM Computing Surveys (CSUR)*, 55(2):1–40, 2022.
- <span id="page-16-7"></span>[27] Samuel G Finlayson, Paea LePendu, and Nigam H Shah. Building the graph of medicine from millions of clinical narratives. *Scientific data*, 1(1):1–9, 2014.
- <span id="page-16-8"></span>[28] Maya Rotmensch, Yoni Halpern, Abdulhakim Tlimat, Steven Horng, and David Sontag. Learning a health knowledge graph from electronic medical records. *Scientific reports*, 7(1):5994, 2017.
- <span id="page-16-9"></span>[29] Krisztian Balog and Tom Kenter. Personal knowledge graphs: A research agenda. In *Proceedings of the 2019 ACM SIGIR International Conference on Theory of Information Retrieval*, pages 217–220, 2019.
- <span id="page-16-13"></span>[30] Kevin B. Johson, Wei-Q Wei, Dilhan Weeraratne, Mark E. Frisse, Karl Misulis, Kyu Rhee, Juan Zhao, and Jane L. Snowdon. Precision medicine, ai, and the future of personalized health care. *Clinical Translational Science*, pages 14(1):86–93, 01 2021.
- <span id="page-16-14"></span>[31] Anna Bauer-Mehren, Michael Rautschka, Ferran Sanz, and Laura I. Furlong. Disgenet: a cytoscape plugin to visualize, integrate, search and analyze gene–disease networks. *Bioinformatics*, 26(22):2924–2926, 09 2010.
- <span id="page-16-15"></span>[32] Yinyin Yuan, Richard Savage, and Florian Markowetz. Patient-specific data fusion defines prognostic cancer subtypes. *PLoS Computational Biology*, page 7(10): e1002227, 2011.
- <span id="page-16-16"></span>[33] Wullianallur Raghupathi and Viju Raghupathi. Big data analytics in healthcare: Promise and potential. *Health Information Science and Systems*, 2:3, 02 2014.
- <span id="page-16-17"></span>[34] Da Xu, Chuanwei Ruan, Evren Korpeoglu, Sushant Kumar, and Kannan Achan. Product knowledge graph embedding for e-commerce. *Proceedings of the 13th International Conference on Web Search and Data Mining*, 01 2020.
- <span id="page-16-18"></span>[35] Sirui Li, Kevin Kok Wai Wong, Lance Chun Che Fung, and Dengya Zhu. Improving question answering over knowledge graphs using graph summarization. *International Conference on Neural Information Processing*, 2022.
- <span id="page-16-19"></span>[36] Kai Liu, Fei Wang, Zhaoyun Ding, Sheng Liang, Zhengfei Yu, and Yun Zhou. A review of knowledge graph application scenarios in cyber security. *ArXiv*, abs/2204.04769, 2022.
- <span id="page-16-20"></span>[37] Qi He, Jaewon Yang, and Baoxu Shi. Constructing knowledge graph for social networks in a deep and holistic way. *Companion Proceedings of the Web Conference 2020*, page 307–308, 2020.

- <span id="page-16-21"></span>[38] J. Puustjarvi and L. Puustjarvi. Providing relevant health information to patient-centered healthcare. In *The 12th IEEE International Conference on e-Health Networking, Applications and Services*, pages 215–220. IEEE, 2010.
- <span id="page-16-22"></span>[39] M. D. Iannacone, S. Bohn, G. C. Nakamura, J. Gerth, K. M. T. Huffer, R. Bridges, E. M. Ferragut, and J. Goodall. Developing an ontology for cyber security knowledge graphs. In *Proceedings of the 10th Annual Cyber and Information Security Research Conference*, pages 1–4. ACM, 2015.
- <span id="page-16-23"></span>[40] L. Sun, M. Yamin, C. Mushi, K. Liu, M. Alsaigh, and F. Chen. Information analytics for healthcare service discovery. *Journal of healthcare engineering*, 5(4):457–477, 2014.
- <span id="page-16-24"></span>[41] Mohammad M. Taye. Understanding semantic web and ontologies: Theory and applications. *Journal of Computing*, 2-6, 6 2010.
- <span id="page-16-25"></span>[42] Natalya F. Noy and Deborah L. McGuinness. Ontology development 101: A guide to creating your first ontology. [https://protege.](https://protege.stanford.edu/publications/ontology_development/ontology101.pdf) [stanford.edu/publications/ontology](https://protege.stanford.edu/publications/ontology_development/ontology101.pdf) development/ontology101.pdf, Mar 2023. Accessed: 2023-07-27.
- <span id="page-16-26"></span>[43] Bhaskar Kapoor and Savita Sharma. A comparative study ontology building tools for semantic web applications. *International journal of Web & Semantic Technology (IJWesT)*, 1(3):1–13, 2010.
- <span id="page-16-27"></span>[44] S. El-Sappagh, F. Franda, F. Ali, and K. Kwak. Snomed ct standard ontology based on the ontology for general medical science. *BMC Medical Informatics and Decision Making*, 18:76, 2018.
- <span id="page-16-28"></span>[45] Josh Hanna et al. Building a drug ontology based on rxnorm and other sources. *Journal of Biomedical Semantics*, 4(1):44, 2013.
- <span id="page-16-29"></span>[46] Chukwuemeka Uchegbu and Xia Jing. The potential adoption benefits and challenges of LOINC codes in a laboratory department: a case study. *Health Information Science and Systems*, 5(1):6, December 2017.
- <span id="page-16-30"></span>[47] Maricela Bravo, Luis Fernando Hoyos Reyes, and Jose A Reyes Ortiz. ´ Methodology for ontology design and construction. *Contadur´ıa y administracion´*, 64(4), 2019.
- <span id="page-16-31"></span>[48] G. Jiang, K. Ogasawara, A. Endoh, and T. Sakurai. Context-based ontology building support in clinical domains using formal concept analysis.*International Journal of Medical Informatics*, 71(1):71–81, 2003.
- <span id="page-16-32"></span>[49] R. Djedidi and M.-A. Aufaure. Medical domain ontology construction approach: A basis for medical decision support. In *Twentieth IEEE International Symposium on Computer-Based Medical Systems (CBMS'07)*, pages 509–511. IEEE, 2007.
- <span id="page-16-33"></span>[50] D. Mendes, I. Rodrigues, and C. Fernandes Baeta. Development and population of an elaborate formal ontology for clinical practice knowledge representation. In *Proceedings of the International Conference on Knowledge Engineering and Ontology Development (IC3K 2013) - KEOD*, pages 286–292. SciTePress, 2013.
- <span id="page-16-34"></span>[51] D. Ognyanov and A. Kiryakov. Tracking changes in rdf(s) repositories. In A. Gomez-P ´ erez and V. R. Benjamins, editors, ´ *Knowledge Engineering and Knowledge Management: Ontologies and the Semantic Web. EKAW 2002. Lecture Notes in Computer Science*, volume 2473, pages 373–378. Springer, Berlin, Heidelberg, 2002.
- <span id="page-16-35"></span>[52] K. Clarkson, A. L. Gentile, D. Gruhl, P. Ristoski, J. Terdiman, and S. Welch. User-centric ontology population. In A. Gangemi et al., editors, *The Semantic Web. ESWC 2018. Lecture Notes in Computer Science*, volume 10843, pages 112–127. Springer, Cham, 2018.
- <span id="page-16-36"></span>[53] S. Ferilli. Integration strategy and tool between formal ontology and graph database technology. *Electronics*, 2021.
- <span id="page-16-37"></span>[54] L. Li, J. Zhao, L. Hou, Y.-k. Zhai, J. Shi, and F. Cui. An attention-based deep learning model for clinical named entity recognition of chinese electronic medical records. *BMC Medical Informatics and Decision Making*, 19, 2019.
- <span id="page-16-38"></span>[55] S. Keretna, C. Peng Lim, and D. Creighton. Enhancement of medical named entity recognition using graph-based features. In *2015 IEEE International Conference on Systems, Man, and Cybernetics*, pages 1895–1900, 2015.
- <span id="page-16-39"></span>[56] Qingchuan Wang and Haihong E. Bi-directional joint embedding of encyclopedic knowledge and original text for chinese medical named entity recognition. In *2021 2nd International Conference on Electronics, Communications and Information Technology (CECIT)*, pages 304–309. IEEE, 2021.
- <span id="page-16-40"></span>[57] Q. Wang, Y. Xia, Y. Zhou, T. Ruan, D. Gao, and P. He. Incorporating dictionaries into deep neural networks for the chinese clinical named entity recognition. *Journal of Biomedical Informatics*, 92, 2018.

- <span id="page-17-0"></span>[58] F. Guan and T. Tezuka. A medical q&a system with entity linking and intent recognition. In *2022 IEEE Symposium Series on Computational Intelligence (SSCI)*, pages 820–829, 2022.
- <span id="page-17-1"></span>[59] Baohang Zhou, Xiangrui Cai, Ying Zhang, Wu Guo, and Xiaojie Yuan. Mtaal: Multi-task adversarial active learning for medical named entity recognition and normalization. In *Proceedings of the AAAI Conference on Artificial Intelligence*, pages 14586–14593, 2021.
- <span id="page-17-2"></span>[60] C. Ruan, Y. Wu, G. S. Luo, Y. Yang, and P. Ma. Relation extraction for chinese clinical records using multi-view graph learning. *IEEE Access*, 8:215613–215622, 2020.
- <span id="page-17-3"></span>[61] A. Li, X. Wang, W. Wang, A. Zhang, and B. Li. A survey of relation extraction of knowledge graphs. In *Web and Big Data. APWeb-WAIM 2019. Lecture Notes in Computer Science*, pages 52–66, 2019.
- <span id="page-17-4"></span>[62] H. Wang, A. Wang, F. Su, H. Feng, and Y. Chen. Construction of therapy-disease knowledge graph (tdkg) based on entity relationship extraction. In *2021 4th International Conference on Advanced Electronic Materials, Computers and Software Engineering (AEMCSE)*, pages 851–855. IEEE, 2021.
- <span id="page-17-5"></span>[63] Q. Zhao, J. Li, C. Xu, J. Yang, and L. Zhao. Knowledge-enhanced relation extraction for chinese emrs. *IT Professional*, 22:57–62, 2020.
- <span id="page-17-6"></span>[64] X. Lv, Y. Guan, J. Yang, and J. Wu. Clinical relation extraction with deep learning. *International Journal of Hybrid Information Technology*, 9:237–248, 2016.
- <span id="page-17-7"></span>[65] V. E. Ghosh and A. Gilboa. What is a memory schema? a historical perspective on current neuroscience literature. *Neuropsychologia*, 53:104–114, 2014.
- <span id="page-17-8"></span>[66] A. Kranjec, G. Ianni, and A. Chatterjee. Schemas reveal spatial relations to a patient with simultanagnosia. *Cortex*, 49:1983–1988, 2013.
- <span id="page-17-9"></span>[67] Y. Ohira, T. Hochin, and H. Nomiya. Introducing specialization and generalization to a graph-based data model. In *Knowledge-Based and Intelligent Information and Engineering Systems. KES 2011. Lecture Notes in Computer Science*, pages 1–13, Berlin, Heidelberg, 2011. Springer.
- <span id="page-17-10"></span>[68] A. Gilboa and H. Marlatte. Neurobiology of schemas and schemamediated memory. *Trends in Cognitive Sciences*, 21:618–631, 2017.
- <span id="page-17-11"></span>[69] Q. Ji, G. Qi, H. Gao, and T. Wu. Survey on schema induction from knowledge graphs. In *Communications in Computer and Information Science*, pages 136–142, 2018.
- <span id="page-17-12"></span>[70] V. G. Aguilar-Escobar, P. Garrido-Vega, and Mar´ıa del Mar Gonzalez- ´ Zamora. Applying the theory of constraints to the logistics service of medical records of a hospital. *European Research on Management and Business Economics*, 22(3):139–146, 2016.
- <span id="page-17-13"></span>[71] G. Mathew and Z. Obradovic. Constraint graphs as security filters for privacy assurance in medical transactions. In *Proceedings of the 2011 ACM Symposium on Applied Computing*, pages 502–504. ACM, 2011.
- <span id="page-17-14"></span>[72] T. Matsuto and S. Shiina. Laboratory test scheduling based on constraint propagation. In *Proceedings of the Twelfth Annual International Conference of the IEEE Engineering in Medicine and Biology Society*, pages 1334–1335. IEEE, 1990.
- <span id="page-17-15"></span>[73] Y. Wu, J. Pan, P. Lu, K. Lin, and X. Yu. Knowledge graph embedding translation based on constraints. *J. Inf. Hiding Multim. Signal Process.*, 8:1119–1131, 2017.
- <span id="page-17-16"></span>[74] Z. H. Pugh and D. Gillan. Propositional constraint graphs: An intuitive, domain-general tool for diagramming knowledge, assumptions, and uncertainties. In *Proceedings of the Human Factors and Ergonomics Society Annual Meeting*, volume 64, pages 254–258. SAGE Publications, 2020.
- <span id="page-17-17"></span>[75] T. K. Tu. A constraint propagation approach for identifying biological pathways in covid-19 knowledge graphs. In *2021 IEEE International Conference on Big Data (Big Data)*, pages 2678–2684. IEEE, 2021.
- <span id="page-17-18"></span>[76] X. Wilcke, M. D. Kleijn, V. D. Boer, H. Scholten, and F. van Harmelen. Bottom-up discovery of context-aware quality constraints for heterogeneous knowledge graphs, 2020.
- <span id="page-17-19"></span>[77] S. Wang, X. Chang, X. Li, G. Long, L. Yao, and Q. Z. Sheng. Diagnosis code assignment using sparsity-based disease correlation embedding. *IEEE Transactions on Knowledge and Data Engineering*, 28:3191– 3202, 2016.
- <span id="page-17-20"></span>[78] S. Ji, S. Pan, E. Cambria, P. Marttinen, and P. S. Yu. A survey on knowledge graphs: Representation, acquisition, and applications. *IEEE Transactions on Neural Networks and Learning Systems*, 33:494–514, 2020.
- <span id="page-17-21"></span>[79] J. Yuan, Z. Jin, H. Guo, H. Jin, X. Zhang, T. H. Smith, and J. Luo. Constructing biomedical domain-specific knowledge graph with mini-

mum supervision. *Knowledge and Information Systems*, 62:317–336, 2019.

- <span id="page-17-22"></span>[80] H. Yu, H. Li, D. Mao, and Q. Cai. A relationship extraction method for domain knowledge graph construction. *World Wide Web*, 23:735–753, 2020.
- <span id="page-17-23"></span>[81] J. Chu, J. Chen, X. Chen, W. Dong, J. Shi, and Z. Huang. Knowledgeaware multi-center clinical dataset adaptation: Problem, method, and application. *Journal of Biomedical Informatics*, page 103710, 2021.
- <span id="page-17-24"></span>[82] J. M. Tomczak and A. Gonczarek. Decision rules extraction from data stream in the presence of changing context for diabetes treatment. *Knowledge and Information Systems*, 34:521–546, 2012.
- <span id="page-17-25"></span>[83] Y. Shang, Y. Tian, M. Zhou, T. Zhou, K. Lyu, Z. Wang, R. Xin, T. Liang, S. Zhu, and J. Li. Ehr-oriented knowledge graph system: Toward efficient utilization of non-used information buried in routine clinical practice. *IEEE Journal of Biomedical and Health Informatics*, 25:2463–2475, 2021.
- <span id="page-17-26"></span>[84] W. Alzoubi. Mining medical databases using graph-based association rules. *International Journal of Machine Learning and Computing*, 2013.
- <span id="page-17-27"></span>[85] N. Ahmadi, T.-T.-D. Truong, L.-H.-M. Dao, S. Ortona, and P. Papotti. Rulehub. *Journal of Data and Information Quality (JDIQ)*, 12:1–22, 2020.
- <span id="page-17-28"></span>[86] A. Wright, J. Pang, J. Feblowitz, F. Maloney, A. Wilcox, H. Ramelson, L. Schneider, and D. Bates. A method and knowledge base for automated inference of patient problems from structured data in an electronic medical record. *Journal of the American Medical Informatics Association: JAMIA*, 18(6):859–867, 2011.
- <span id="page-17-29"></span>[87] M. Brundage, K. Smith, E. Little, E. Bantug, and C. Snyder. Communicating patient-reported outcome scores using graphic formats: results from a mixed-methods evaluation. *Quality of Life Research*, 24:2457– 2472, 2015.
- <span id="page-17-30"></span>[88] S. Issa, O. Adekunle, F. Hamdi, S. Cherfi, M. Dumontier, and A. Zaveri. Knowledge graph completeness: A systematic literature review. *IEEE Access*, 9:31322–31339, 2021.
- <span id="page-17-31"></span>[89] N. Sharma and R. Bhatt. Privacy preserving knowledge graph for healthcare applications. *Journal of Physics: Conference Series*, 2339, 2022.
- <span id="page-17-32"></span>[90] L. Li, P. Wang, J. Yan, Y. Wang, S. Li, J. Jiang, Z. Sun, B. Tang, T.- H. Chang, S. Wang, and Y. Liu. Real-world data medical knowledge graph: construction and applications. *Artificial Intelligence in Medicine*, 103:101817, 2020.
- <span id="page-17-33"></span>[91] A. Talukder, L. Schriml, A. Ghosh, R. Biswas, P. Chakrabarti, and R. E. Haas. Diseasomics: Actionable machine interpretable disease knowledge at the point-of-care. *PLOS Digital Health*, 1, 2022.
- <span id="page-17-34"></span>[92] M. Rospocher, M. van Erp, P. Vossen, A. Fokkens, I. Aldabe, G. Rigau, and T. Bogaard. Building event-centric knowledge graphs from news. *Journal of Web Semantics*, 37-38:132–151, 2016.
- <span id="page-17-35"></span>[93] Zhisheng Huang, Jie Yang, Frank van Harmelen, and Qing Hu. Constructing disease-centric knowledge graphs: a case study for depression (short version). In *Artificial Intelligence in Medicine: 16th Conference on Artificial Intelligence in Medicine, AIME 2017, Vienna, Austria, June 21-24, 2017, Proceedings 16*, pages 48–52. Springer, 2017.
- <span id="page-17-36"></span>[94] Z. Huang, J. Yang, F. V. Harmelen, and Q. Hu. Constructing disease-centric knowledge graphs: A case study for depression (short version). In *Artificial Intelligence in Medicine*, pages 48–52. Springer International Publishing, 2017.
- <span id="page-17-37"></span>[95] Enayat Rajabi and Somayeh Kafaie. Knowledge graphs and explainable ai in healthcare. *Inf.*, 13(459), 2022.
- <span id="page-17-38"></span>[96] Wenying Yu, Shuai Ding, Zijie Yue, and Shanlin Yang. Emotion recognition from facial expressions and contactless heart rate using knowledge graph. In *2020 IEEE International Conference on Knowledge Graph (ICKG)*, pages 64–69. IEEE, 2020.
- <span id="page-17-39"></span>[97] Ying Ding, Bogdan Arsintescu, Ching-Hua Chen, Haoyun Feng, Franc¸ois Scharffe, Oshani Seneviratne, and Juan Sequeda. International workshop on knowledge graph: Heterogenous graph deep learning and applications. In *Proceedings of the 27th ACM SIGKDD Conference on Knowledge Discovery & Data Mining*, page 4121–4122, 2021.
- <span id="page-17-40"></span>[98] M. D. Lytras, E. Sakkopoulos, and P. O. Pablos. Semantic web and knowledge management for the health domain: state of the art and challenges for the seventh framework programme (fp7) of the european union (2007-2013). *Int. J. Technol. Manag.*, 47:239–249, 2009.
- <span id="page-17-41"></span>[99] X. Wang, Y. Wang, C. Gao, K. Lin, and Y. Li. Automatic diagnosis with efficient medical case searching based on evolving graphs. *IEEE Access*, 6:53307–53318, 2018.

- <span id="page-18-0"></span>[100] L. Plaza and A. D´ıaz. Retrieval of similar electronic health records using umls concept graphs. In *Natural Language Processing and Information Systems*, pages 296–303, 2010.
- <span id="page-18-1"></span>[101] Y. Liang, H. Wang, and W. Zhang. Deep knowledge reasoning guided disease prediction. In *2022 IEEE International Conference on Systems, Man, and Cybernetics (SMC)*, pages 2695–2702, 2022.
- <span id="page-18-2"></span>[102] N. Heilig, J. Kirchhoff, F. Stumpe, J. Plepi, L. Flek, and H. Paulheim. Refining diagnosis paths for medical diagnosis based on an augmented knowledge graph. *ArXiv*, abs/2204.13329, 2022.
- <span id="page-18-3"></span>[103] Y. Li, B. Qian, X. Zhang, and H. Liu. Graph neural network-based diagnosis prediction. *Big data*, 2020.
- <span id="page-18-4"></span>[104] S. Zheng, Z. Zhu, Z. Liu, Z. Guo, Y. Liu, and Y. Zhao. Multi-modal graph learning for disease prediction. *IEEE Transactions on Medical Imaging*, 41:2207–2216, 2021.
- <span id="page-18-5"></span>[105] Z. Sun, W. Dong, J. Shi, and Z. Huang. Interpretable disease prediction based on reinforcement path reasoning over knowledge graphs. *ArXiv*, abs/2010.08300, 2020.
- <span id="page-18-6"></span>[106] Z. Sun, H. Yin, H. Chen, T. Chen, L. Cui, and F. Yang. Disease prediction via graph neural networks. *IEEE Journal of Biomedical and Health Informatics*, 25:818–826, 2020.
- <span id="page-18-7"></span>[107] P. Liu, X. Wang, X. Sun, X. Shen, X. Chen, Y. Sun, and Y. Pan. Hkdp: A hybrid knowledge graph based pediatric disease prediction system. In *Smart Health*, pages 78–90. Springer International Publishing, 2017.
- <span id="page-18-8"></span>[108] K. Zhang. A knowledge graph based medical intelligent question answering system. In *2021 IEEE International Conference on Computer Science, Electronic Information Engineering and Intelligent Control Technology (CEI)*, pages 460–465, 2021.
- <span id="page-18-9"></span>[109] B. Saha, V. Mandloi, and S. Ghosh. Predicting missing and noisy links via neighbourhood preserving graph embeddings in a clinical knowlegebase. In *2020 19th IEEE International Conference on Machine Learning and Applications (ICMLA)*, pages 1124–1131, 2020.
- <span id="page-18-10"></span>[110] S. Biswas, P. Mitra, and K. S. Rao. Relation prediction of co-morbid diseases using knowledge graph completion. *IEEE/ACM Transactions on Computational Biology and Bioinformatics*, 18:708–717, 2019.
- <span id="page-18-11"></span>[111] L. Wang, H. Xie, W. Han, X. Yang, L. Shi, J. Dong, K. Jiang, and H. Wu. Construction of a knowledge graph for diabetes complications from expert-reviewed clinical evidences. *Computer Assisted Surgery*, 25:29–35, 2020.
- <span id="page-18-19"></span>[112] J. Dorpinghaus, T. H ¨ ubenthal, and J. Faber. A novel link prediction ¨ approach on clinical knowledge graphs utilising graph structures. In *2022 17th Conference on Computer Science and Intelligence Systems (FedCSIS)*, pages 43–52, 2022.
- <span id="page-18-12"></span>[113] A. Gyrard, M. Gaur, K. Thirunarayan, A. Sheth, and S. Shekarpour. Personalized health knowledge graph. *CEUR Workshop Proceedings*, 2317, 2018.
- <span id="page-18-13"></span>[114] P. Ping, K. Watson, J. Han, and A. Bui. Individualized knowledge graph: A viable informatics path to precision medicine. *Circulation Research*, 120:1078–1080, 2017.
- <span id="page-18-14"></span>[115] Q. Zhu, C. Qu, R. Liu, G. Vatas, A. Clough, D. Nguyen, E. Sid, E. Mathe, and Y. Xu. Rare disease-based scientific annotation knowl- ´ edge graph. *Frontiers in Artificial Intelligence*, 5, 2022.
- <span id="page-18-15"></span>[116] Y. Xie, C. Yan, and D. Zhang. Personalized diagnostic modal discovery of traditional chinese medicine knowledge graph. In *2018 14th International Conference on Natural Computation, Fuzzy Systems and Knowledge Discovery (ICNC-FSKD)*, pages 1096–1103, 2018.
- <span id="page-18-16"></span>[117] X. Li, G. Tur, D. Z. Hakkani-T ¨ ur, and Q. Li. Personal knowledge graph ¨ population from user utterances in conversational understanding. In *2014 IEEE Spoken Language Technology Workshop (SLT)*, pages 224– 229, 2014.
- <span id="page-18-17"></span>[118] Z. Zhang, P. Yu, N. Pai, H. Chang, S. Chen, M. Yin, T. Song, S. Lau, and C. Deng. Developing an intuitive graph representation of knowledge for nonpharmacological treatment of psychotic symptoms in dementia. *Journal of Gerontological Nursing*, 48(4):49–55, 2022.
- <span id="page-18-18"></span>[119] L. Gortzis and G. Nikiforidis. Tracing and cataloguing knowledge in an e-health cardiology environment. *Journal of Biomedical Informatics*, 41(2):217–223, 2008.
- <span id="page-18-20"></span>[120] X. Xiang, Z. Wang, Y. Jia, and B. Fang. Knowledge graph-based clinical decision support system reasoning: A survey. In *2019 IEEE Fourth International Conference on Data Science in Cyberspace (DSC)*, pages 373–380, 2019.
- <span id="page-18-21"></span>[121] D. N. Nicholson and C. Greene. Constructing knowledge graphs and their biomedical applications. *Computational and Structural Biotechnology Journal*, 18:1414–1428, 2020.

- <span id="page-18-22"></span>[122] M. Wang, M. Liu, J. Liu, S. Wang, G. Long, and B. Qian. Safe medicine recommendation via medical knowledge graph embedding. *ArXiv*, abs/1710.05980, 2017.
- <span id="page-18-23"></span>[123] N. Sharma. Patient-centric approach for clinical trials: Current trend and new opportunities. *Perspectives in Clinical Research*, 6:134–138, 2015.
- <span id="page-18-24"></span>[124] H. Weng, Z. Liu, S. Yan, M. Fan, A. Ou, D. Chen, and T. Hao. A framework for automated knowledge graph construction towards traditional chinese medicine. In *HealthInformation Science*, volume 10594, pages 170–181. Springer International Publishing, 2017.
- <span id="page-18-25"></span>[125] N. Rastogi and M. J. Zaki. Personal health knowledge graphs for patients. *ArXiv*, abs/2004.00071, 2020.
- <span id="page-18-26"></span>[126] Q. Cong, Z. Feng, F. Li, L. Zhang, G. Rao, and C. Tao. Constructing biomedical knowledge graph based on semmeddb and linked open data. In *2018 IEEE International Conference on Bioinformatics and Biomedicine (BIBM)*, pages 1628–1631. IEEE, 2018.
- <span id="page-18-27"></span>[127] L. Hu, M. Zhang, S. Li, J. Shi, C. Shi, C. Yang, and Z. Liu. Textgraph enhanced knowledge graph representation learning. *Frontiers in Artificial Intelligence*, 4, 2021.
- <span id="page-18-28"></span>[128] Y. Zhang, M. Sheng, R. Zhou, Y. Wang, G. Han, H. Zhang, C. Xing, and J. Dong. Hkgb: An inclusive, extensible, intelligent, semi-autoconstructed knowledge graph framework for healthcare with clinicians' expertise incorporated. *Inf. Process. Manag.*, 57:102324, 2020.
- <span id="page-18-29"></span>[129] Mishall Al-Zubaidie, Zhongwei Zhang, and Ji Zhang. Pax: Using pseudonymization and anonymization to protect patients' identities and data in the healthcare system. *International Journal of Environmental Research and Public Health*, 2019.
- <span id="page-18-30"></span>[130] S. Hooshafza, F. Orlandi, R. Flynn, L. McQuaid, G. Stephens, and L. O'Connor. Modelling temporal data in knowledge graphs: a systematic review protocol. *HRB Open Research*, 2021.
- <span id="page-18-31"></span>[131] J. Wu. Construct a knowledge graph for china coronavirus (covid-19) patient information tracking. *Risk Management and Healthcare Policy*, 14:4321–4337, 2021.
- <span id="page-18-32"></span>[132] P. Chandak, K. Huang, and M. Zitnik. Building a knowledge graph to enable precision medicine. *Scientific data*, 10, 2023.
- <span id="page-18-33"></span>[133] C. Peng, F. Xia, M. Naseriparsa, and F. Osborne. knowledge graphs: opportunities and challenges. *Artificial Intelligence Review*, 56:13071– 13102, 2023.
- <span id="page-18-34"></span>[134] P. Solanki, J. Grundy, and W. Hussain. Operationalising ethics in artificial intelligence for healthcare: a framework for ai developers. *ai and ethics*, 3:223–240, 2022.
- <span id="page-18-35"></span>[135] S. Maxhelaku, E. Xhumari, and E. Xhina. Exploiting knowledge graphs in healthcare. In *Proceedings of the 16th Economics & Finance*, pages 111–118, 2022.
